--- 
title: 'Genome annotation'
output:
  bookdown::html_document2: 
    toc: TRUE
    toc_depth: 4
    toc_float: TRUE
link-citations: yes
fontsize: 12pt
bibliography: References.bib
---

```{r packages, echo=FALSE, warning=FALSE, include=FALSE}
library(knitr)
library(rmarkdown)
library(bookdown)
#library(distill)
library(knitcitations)
library(formatR)
library(devtools)
library(usethis)

#Generate BibTex citation file for all R packages used to produce report
knitr::write_bib(.packages(), file = 'packages.bib')
```


```{r setup, include=FALSE, cache=FALSE, message = FALSE}
#Load packages
library(knitr)
library(rmarkdown)
# Chunk options: see http://yihui.name/knitr/options/ ###

## Text results
opts_chunk$set(echo = TRUE, warning = TRUE, message = TRUE, include = TRUE)

## Code decoration
#opts_chunk$set(tidy = TRUE, R.options = list(width = 60), highlight = TRUE)

## Caching code
opts_chunk$set(cache = 2, cache.path = "cache/")

## Plots
#opts_chunk$set(fig.path = "Figures_MS/", dev=c('pdf', 'png'), dpi = 300)

## Locate figures as close as possible to requested position (=code)
#opts_chunk$set(fig.pos = "H")
```

# Overview

To annotate the sagebrush genome (and identify adaptive loci related to drought and heat stress) a suite of GxE experiments are conducted followed by RNA sequencing and transcriptome assemblies (later referred to as bioinformatic analyses). This document is dedicated to describing our experimental design, which is associated to comparing drought and heat performance between sagebrush individual lines and identifying adaptive loci associated to these phenotypes. Leaf and root tissues will be sequenced using RNA-seq technology (on Illumina NovaSeq platform). 

# Methodological approach

Our project aims at assessing genome to phenome mechanisms involved in drought (D), heat (H) and combined drought + heat (DH) tolerance in two sagebrush individual lines originating from contrasting climates (and corresponding to different genotypes). In addition, this approach will allow quantifying and comparing genotype responses to acute abiotic stress therefore supporting forecasting efforts to evaluate the adaptive potential of sagebrush populations to climate change. To reach these goals, we are applying a 3-tier approach:

1. Conduct GxE experiments on individual lines (= genotypes) and monitor vitals associated to mechanisms leading to stomatal closure. This phenotype is key to avoid seedlings' mortality by cavitation or embolism during acute abiotic stress.
2. Assess treatments effect by targeting transcripts associated to the ABA pathway resulting into stomatal closure. These data will allow quantifying and comparing responses of individual lines to treatments.
3. Discover adaptive loci critical to D, H, DH tolerance by comparing control and treatment transcriptomes between and within individual lines. A custom RNA baiting kit will be developed to survey adaptive loci across populations to estimate their adaptive potential to cope with climate change.

Descriptions of the three stages of this approach is provided below.

# Conduct GxE experiments

## Individual lines

GxE experiments were conducted on `G1_b2` (from ID representing drought tolerant genotype, but undergoing low summer heat stress) and `G2_b24` (from UT representing drought sensitive genotype, but undergoing high summer heat stress) individual lines. 30 plantlets were generated per individual line per treatment (n=30) with 5 plantlets for associated control treatments (n=5). Please see Figure \@ref(fig:timetab) for more details (see below for more details). Plantlets were grown in PhytoCon 946 ml vessels with XX gr of soil (4 sand: 1 vermiculite ratio) and acclimitized to condtions in the P4 Percival culture chamber. One week prior to imposing treatment a sealed plastic lid covering the soil was applied to prevent evaporation and only support water movement via transpiration. This setting was key to infer timing of stomatal closure as a response to applied treatments (see below for more details). 

## Vital measurements

Soil moisture, water potential ($\Psi~soil~$) and air relative humidity were monitored during treatments to assess plantlets transpiration rates and evaluate stomatal closure using XXX devices. In the case of D and DH treatments, plantlets were harvested when stomata were fully closed 9meaning that they are entering into starvation to avoid death by cavitation; see below for more details). Leaf sizes were inadequate to measure stomatal conductance and we therefore had to apply a different approach to measure this process. This was estimated by comparing night vs. day soil moisture data using a t-test (here we were after H0 = no significant difference between the two datasets). Upon stomatal closure, transpiration would stop therefore soil moisture should remain identical between night and day. This assumption is only correct when evaporation is null and that variation in soil moisture is only driven by transpiration.  

Estimating this process will be facilitated by the lid preventing evaporation (see above). To further estimate cavitation upon stomatal closure stem xylem pressure ($\Psi~x~$) was measured when plantlets were harvested using the XXX device. This trait corresponds to the plant's ability to undergo high xylem pressure caused by drought resulting in air bubbles entering the xylem and disrupting the water flow therefore causing cavitation. To avoid death by cavitation, plants evolved mechanisms to close their stomata, which are mostly regulated by ABA. The ability of a plant to sustain high xylem pressure and therefore minimize cavitation can be predicted by its vessels' pore diameters. In this context, stem and root xylem anatomies of our individual lines were quantified and compared using SEM and light microscopy. Finally, transpiration rates have been shown to be associated to stomata density. Stomata density on both side of the blade were quantified and compared between individual lines using SEM.


**GxE experiments:** Drought (D), Heat (H) and drought + heat (DH) treatments were conducted on 30 plantlets per individual line per treatment. Treatments were conducted in the P4 Percival culture chamber were all plantlets were distributed on one shelf to avoid any light or temperature effects. Five control (C) plantlets associated to each treatment were harvested at 0 d (just before applying the treatment) to enable assessing treatment efficacy and identifying differentially expressed genes (DEGs) in leaves and roots using RNA-seq technique.  


**Transcriptomics:** Transcriptomic analyses were conducted on leaf and root tissues. Since all individuals are clones, RNA extractions were conducted on 5 randomly selected plantlets and pooled (by tissue and treatment) to conduct Illumina NovaSeq RNA-seq. Controls associated to each treatment (D, H, DH) were kept distinct (= distinct libraries) to allow for accurate comparisons and identifications of DEGs. In this context, a total of 12 Illumina libraries were built and sequenced on 2 Illumina NovaSeq runs (= 6 libraries per run).

## Sampling

[Individual lines](Individual_lines.html) that have been acclimatized to natural environmental conditions are used for these experiments. Every plantlet within an individual line will have a specific barcode to monitor its progress. Two individual lines will be used: `G1_b2` (representing drought tolerant genotype) and `G2_b24` (representing drought sensitive genotype). We will be conducting trial GxE experiments using `G2_b27` to make sure that all the settings are correct. Please see Figure \@ref(fig:timetab) for more details.

## Timetable

The timetable associated to the GxE experiments and subsequent RNA sequencing (RNA-seq) and bioinformatic analyses (= transcriptome assemblies) required to annotate the sagebrush genome is presented in Figure \@ref(fig:timetab).

The time allocated for each phase of the workflow (and representing the time associated to process an individual line) is as follows:

- <span style="color: black;font-weight:bold">Ex planting</span>: 11 weeks. Note that the first two weeks and the final week of this phase are taking place in the culture chamber (P4). Please see the [Propagation](Individual_lines.html) for more details.
- <span style="color: grey;font-weight:bold">GxE experiments</span>: 4 weeks. This time should allow for conducting the three GxE experiments (T~H~, T~D~, T~DH~) back to back on one individual line. This will take place in the culture chamber (P4).
- <span style="color: green;font-weight:bold">RNA-seq</span>: 10 weeks. Two weeks are dedicated to conduct RNA extractions at BSU and shipping samples to sequencing facility. We estimate that eight weeks will be required to conduct the sequencing.
- <span style="color: blue;font-weight:bold">Bioinformatics</span>: 20 weeks. Assembling transcriptomes, conducting comparative analyses to isolate adaptive loci and starting genome annotation will take 20 weeks.  


```{r timetab, echo=F, warning=F, fig.cap= "Timetable associated to GxE experiments and subsequent RNA sequencing and biorinformatic analyses used for annotate the sagebrush genome. Please see text for more details on individual lines."}
#Timetable

timetab <- data.frame("Individual_line" = c("G2_b27", "G2_b24", "G1_b2"), "Start_ExPlanting" = lubridate::mdy(c("NA", "NA", "NA")), "End_ExPlanting" = lubridate::mdy(c("NA", "NA", "NA")), "GxE_End" = lubridate::mdy(c("NA", "NA", "NA")), "RNA-seqEnd" = lubridate::mdy(c("NA", "NA", "NA")), "Bioinfo_End" = lubridate::mdy(c("NA", "NA", "NA")))


###
#Trial
# G2_b27
###

#When ready for ex planting?
timetab$Start_ExPlanting[1] <- as.Date("04-01-21", tryFormats = c("%m-%d-%y")) + lubridate::weeks(2)

#When ready for GxE
timetab$End_ExPlanting[1] <- timetab$Start_ExPlanting[1] + lubridate::weeks(10)

#When GxE end?
timetab$GxE_End[1] <- timetab$End_ExPlanting[1] + lubridate::weeks(4)

###
# G2_b24
###

#When ready for ex planting?
timetab$Start_ExPlanting[2] <- as.Date("04-01-21", tryFormats = c("%m-%d-%y")) + lubridate::weeks(5)

#When ready for GxE
timetab$End_ExPlanting[2] <- timetab$Start_ExPlanting[2] + lubridate::weeks(11)

#When GxE end?
timetab$GxE_End[2] <- timetab$End_ExPlanting[2] + lubridate::weeks(4)

#When RNA-seq end?
timetab$RNA.seqEnd[2] <-  timetab$GxE_End[2] + lubridate::weeks(10)

#When Bioinfo end?
timetab$Bioinfo_End[2] <-  timetab$RNA.seqEnd[2] + lubridate::weeks(20)


###
# G1_b2
###

#When ready for ex planting?
timetab$Start_ExPlanting[3] <- as.Date("04-09-21", tryFormats = c("%m-%d-%y")) + lubridate::weeks(13)

#When ready for GxE
timetab$End_ExPlanting[3] <- timetab$Start_ExPlanting[3] + lubridate::weeks(11)

#When GxE end?
timetab$GxE_End[3] <- timetab$End_ExPlanting[3] + lubridate::weeks(4)

#When RNA-seq end?
timetab$RNA.seqEnd[3] <-  timetab$GxE_End[3] + lubridate::weeks(10)

#When Bioinfo end?
timetab$Bioinfo_End[3] <-  timetab$RNA.seqEnd[3] + lubridate::weeks(20)



###
#PLOT
###

#Plot timetable
plot(x=c(lubridate::floor_date(min(timetab$Start_ExPlanting), "month"), lubridate::round_date(max(timetab$Bioinfo_End, na.rm = T), "month")), y= c(0,3), type="n", axes = F, ylab = "", xlab = "")

#Add customed axes
#Y axis = Individual lines
axis(side = 2, at = c(0.5, 1.5, 2.5), labels = timetab$Individual_line)

#X axis = time
datespan <- seq(from = lubridate::floor_date(min(timetab$Start_ExPlanting), "month"), to = lubridate::round_date(max(timetab$Bioinfo_End, na.rm = T), "month"), "month")

axis(side = 1, at = datespan, labels = as.vector(lubridate::month(datespan, label = TRUE, abbr = T)))

#Add titles to axes
title(main="", xlab="Time (2021-2022)", ylab="Sagebrush individual lines")

for(i in 1:nrow(timetab)){
  #Ex planting
  segments(x0 = as.Date(timetab$Start_ExPlanting[i]), x1 = as.Date(timetab$End_ExPlanting[i]), y0 = i-0.5, y1 = i-0.5, col= "black", lwd = 2)
  #GxE
  segments(x0 = as.Date(timetab$End_ExPlanting[i]), x1 = as.Date(timetab$GxE_End[i]), y0 = i-0.5, y1 = i-0.5, col= "grey", lwd = 2)
  #RNAseq
  segments(x0 = as.Date(timetab$GxE_End[i]), x1 = as.Date(timetab$RNA.seqEnd[i]), y0 = i-0.5, y1 = i-0.5, col= "green", lwd = 2)
  #Bioinfo
  segments(x0 = as.Date(timetab$RNA.seqEnd[i]), x1 = as.Date(timetab$Bioinfo_End[i]), y0 = i-0.5, y1 = i-0.5, col= "blue", lwd = 2)

  
  #Dates
  text(x = c(timetab$Start_ExPlanting[i], timetab$End_ExPlanting[i], timetab$GxE_End[i], timetab$RNA.seqEnd[i], timetab$Bioinfo_End[i]), y = rep(i-0.2, 3), c(timetab$Start_ExPlanting[i], timetab$End_ExPlanting[i], timetab$GxE_End[i], timetab$RNA.seqEnd[i], timetab$Bioinfo_End[i]), srt=90, cex=.5)
}

#Arrows with fiscal years
#Y3
arrows(x0 = lubridate::floor_date(min(timetab$Start_ExPlanting), "month"), x1 = as.Date("10-01-21", tryFormats = c("%m-%d-%y")), y0 = 0.1, y1 = 0.1, length = 0.1, angle = 10)
text(x = as.Date("06-01-21", tryFormats = c("%m-%d-%y")), y = 0.2, "Year 3", cex = 0.8)

#Y4
arrows(x0 = as.Date("10-01-21", tryFormats = c("%m-%d-%y")), x1 = as.Date("06-01-22", tryFormats = c("%m-%d-%y")), y0 = 0.1, y1 = 0.1, length = 0.1, angle = 10, code = 1)
text(x = as.Date("02-01-22", tryFormats = c("%m-%d-%y")), y = 0.2, "Year 4", cex= 0.8)

# Add a legend
legend("bottomright", 
  legend = c("Ex planting (11 w.)", "GxE exp. (4 w.)", "RNA-seq (10 w.)", "Bioinformatics (20 w.)"), 
  col = c("black","grey", "green", "blue"), 
  lwd = rep(2, 4),
  horiz = F , 
  inset = c(0, 0.1),
  cex = 0.6)
```

### Climatic growing conditions

During the ex planting protocol, plantlets are exposed to the following climatic conditions in:

- On the cart in the laboratory: 24<span>&deg and average of 20.3% RH.
- In the culture chamber (P4): 24<span>&deg and average of 22% RH.

## Drought treatment

```{r, droughtplot, echo = FALSE, warning = FALSE, message = FALSE, results = "hide"}
###
#Load R packages
###
require(png)

###
#Load user-defined functions (UDFs)
###
#Create list of R UDFs
files.sources <- list.files("Functions", pattern = ".R", full.names=T)
#Source R UDFS
sapply(files.sources[1], source)

###
#Load stomata logos (PNG)
###
#Located in Images
stopen <- png::readPNG("Images/Stomata_opened.png")
stclosed <- png::readPNG("Images/Stomata_closed.png")
dayLogo <- png::readPNG("Images/sun-128x128.png")
nightLogo <- png::readPNG("Images/Moon.png")

#GxE drought

DroughtTime <- data.frame("Time" = seq(from=0, to=7, by=1),  "Soil_moisture" = c(80,60,60,40,40,20,20,20), "temp" = rep(24, 8))

#Plot
plot(x=DroughtTime$Time, y=DroughtTime$Soil_moisture, ylim=c(0,105), xlab = "", ylab="", type = "n", frame.plot = F, axes = F)

#Add day/night rectangles
nightstart <- seq(from=1, to = 5, by=2)
for(i in 1:length(nightstart)){
  rect(xleft = nightstart[i], ybottom = -10, xright = nightstart[i]+1, ytop = 100, col = "grey", border=NA)
}

#Add logo of stomata
stopenx <- seq(from=0.5, to=4.5, by=2)
for(i in 1:length(stopenx)){
  addImg(stopen, x = stopenx[i], y = 90, width = 0.5)
}

stclosedx <- c(seq(from=1.5, to=7, by=2), 6.5)
for(i in 1:length(stclosedx)){
  addImg(stclosed, x = stclosedx[i], y = 90, width = 0.5)
}

#Add moisture data
lines(x=DroughtTime$Time, y=DroughtTime$Soil_moisture, lwd=2, col="black")

#Add side axis with temperature
lines(x=DroughtTime$Time, y=2*DroughtTime$temp, lwd=2, col="blue")

#Add temperature axis
axis(side=4, at=2*seq(from=0, to=45, by=5), labels = seq(from=0, to=45, by=5), pos=c(7.1,0))
mtext(side=4, line = 1.2, "Temperature (Celsius)", col="blue", font=1, cex=1)

#Add soil moisture axis
axis(side=2, at=seq(from=0, to=100, by=20), labels = seq(from=0, to=100, by=20), pos=c(-0.1,0))
mtext(side=2, line=2.2, "Soil moisture (%)", col="black", font=1, cex=1)

#Add time axis
axis(side=1, at=DroughtTime$Time, labels = DroughtTime$Time/2)
mtext(side=1, line = 2.2, "Time (days)", col="black", font=1, cex=1)

#Add cavitation threshold
segments(x0=0, x1=7, y0=10, y1=10, lty=2)
text(x=0.5, y=11, "Cavitation\nthreshold", cex=0.8)
#Add safety margin
arrows(x0=6.1, x1=6.1, y0=10, y1=20, angle=10, code=3, length = 0.1)
text(x=6.5, y=15, "Safety\nmargin", cex=0.8)

#Add logo of day
dayx <- seq(from=0.5, to=6.5, by=2)
for(i in 1:length(dayx)){
  addImg(dayLogo, x = dayx[i], y = 2, width = 0.4)
}

#Add logo of night
nightx <- seq(from=1.5, to=7, by=2)
for(i in 1:length(nightx)){
  addImg(nightLogo, x = nightx[i], y = 2, width = 0.3)
}

#Add harvesting of tissues
#Controls
text(x=0.1, y=83, expression(paste("RNA-seq,", Psi[x])), srt=90, cex=0.8, adj=c(0,0))
points(x=0, y=80, pch=16, cex=1.5)

#Treated
text(x=6.2, y=22, expression(paste("RNA-seq,", Psi[x])), srt=45, cex=0.8, adj=c(0,0))
points(x=6.2, y=20, pch=16, cex=1.5)

```



**Growing conditions:** at constant 24<span>&deg;</span>C. 

**Vital measureme

**Treatments:**

Two treatments will be conducted to assess the effect of drought on gene expression patterns:

  - T~C~: well-watered (control). 
  - T~D~: drought. 

**Sampling**

Each T~D~ plantlet will be sampled at the onset of drought (upon wilting), whereas T~C~ plantlets will only be sampled at the end of the experiment (when all the T~D~ plantlets are wilted). A response curve will be drawn to estimate phenotypic plasticity within each individual line. Hydraulic conductivity upon sampling will be measured for each plantlet.


#### Heat experiment

**Growing conditions:** at constant 40<span>&deg;</span>C. Based on data from UT2.

42<span>&deg;</span>C for most herbaceous plants:

Tissue collected at 0 h (= control) and 5 h (=heat treatment) for *Brachypodium distachyon*. [See] here(https://www.frontiersin.org/articles/10.3389/fpls.2016.02067/full)

Tissue collected after 2 days for *Brassica alboglabra*. See [here] (https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-019-5652-y)

45<span>&deg;</span>C for a shrub (Ziziphus) for 5 days with tissue sampling on 0d (control), 1d, 3d, 5d. 

Plantlets will be well-watered before conducting the experiment.

**Treatments:**

One treatment will be conducted to assess the effect of heat on gene expression patterns:

  - T~H~: heat treatment. 

**Sampling**

Each T~H~ plantlet will be sampled at the onset of heat (upon chlorosis). A response curve will be drawn to estimate phenotypic plasticity within each individual line. Hydraulic conductivity upon sampling will be measured for each individual.


#### Heat and drought experiment

**Growing conditions:** Start at 24<span>&deg;</span>C and gradually increase to 40<span>&deg;</span>C. Every day, we will increase the temperature by 4 <span>&deg;</span>C and it will take 5 days to reach 40 <span>&deg;</span>C (temperatures: `r seq(from=24, to=40, by=4)`). 

Based on data from UT2.

**Treatments:**

Two treatments will be conducted to assess the effect of drought on gene expression patterns:

  - T~CH~: well-watered (control), but experiencing gradual increase of heat. 
  - T~D+H~: Drought and heat stress. 

**Sampling**

The onset of drought for each plantlet will be recorded (when they start wilting), but tissue will be sampled at chlorosis (= heat stress). We hypothesize that these two phenotypes might co-occur due to the growing conditions. T~D+H~ individuals will be sampled when these show sign of chlorosis (= heat stress). Response curves (associated to drought and heat phenotypes) will be drawn to estimate phenotypic plasticity within each individual line. Hydraulic conductivity upon sampling will be measured for each individual.

## Transcriptomic analyses

## RNA extraction

1. Randomly select 5 plantlet per treatment per individual line to obtain representation of transcriptomic response at individual line.
2. Weigh 10 mg of both leaf and root tissues per randomly selected plantlet. 
3. Pool leaf tissues per treatment per individual line in a tube (= ready for RNA extraction).
4. Pool root tissues per treatment per individual line in a tube (= ready for RNA extraction).
5. Conduct RNA extractions.
6. RNA-seq 


# References

<div id="refs"></div>

# Appendix 1 {.appendix}

Citations of all R packages used to generate this report. 

```{r generateBibliography, eval = T, results="asis", cache = F, echo=F, warning = FALSE, message=FALSE}
library("knitcitations")
cleanbib()
options("citation_format" = "pandoc")
read.bibtex(file = "packages.bib")
``` 

# Appendix 2 {.appendix}

Version information about R, the operating system (OS) and attached or R loaded packages. This appendix was generated using `sessionInfo()`.

```{r eval=T, echo=F, warning = FALSE, message=FALSE}
sessionInfo()
```
