<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Genome annotation</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Sagebrush Genome Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Individual_lines.html">Propagation</a>
</li>
<li>
  <a href="Biomass_production.html">Biomass</a>
</li>
<li>
  <a href="Sequencing.html">Sequencing</a>
</li>
<li>
  <a href="Annotation.html">Annotation</a>
</li>
<li>
  <a href="Protocols.html">Protocols</a>
</li>
<li>
  <a href="Team.html">Team</a>
</li>
<li>
  <a href="funding.html">Acknowledgements</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Genome annotation</h1>

</div>


<div id="overview" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Overview</h1>
<p>To annotate the sagebrush genome (and identify adaptive loci related to drought and heat stress) a suite of GxE experiments are conducted followed by RNA sequencing and transcriptome assemblies (later referred to as bioinformatic analyses). This document is dedicated to describing our experimental design, which is associated to comparing drought and heat performance between sagebrush individual lines and identifying adaptive loci associated to these phenotypes. Leaf and root tissues will be sequenced using RNA-seq technology (on Illumina NovaSeq platform).</p>
</div>
<div id="methodological-approach" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Methodological approach</h1>
<p>Our project aims at assessing genome to phenome mechanisms involved in drought (D), heat (H) and combined drought + heat (DH) tolerance in two sagebrush individual lines originating from contrasting climates (and corresponding to different genotypes). In addition, this approach will allow quantifying and comparing genotype responses to acute abiotic stress therefore supporting forecasting efforts to evaluate the adaptive potential of sagebrush populations to climate change. To reach these goals, we are applying a 3-tier approach:</p>
<ol style="list-style-type: decimal">
<li>Conduct GxE experiments on individual lines (= genotypes) and monitor vitals associated to mechanisms leading to stomatal closure. This phenotype is key to avoid seedlings’ mortality by cavitation or embolism during acute abiotic stress.</li>
<li>Assess treatments effect by targeting transcripts associated to the ABA pathway resulting into stomatal closure. These data will allow quantifying and comparing responses of individual lines to treatments.</li>
<li>Discover adaptive loci critical to D, H, DH tolerance by comparing control and treatment transcriptomes between and within individual lines. A custom RNA baiting kit will be developed to survey adaptive loci across populations to estimate their adaptive potential to cope with climate change.</li>
</ol>
<p>Descriptions of the three stages of this approach is provided below.</p>
</div>
<div id="conduct-gxe-experiments" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Conduct GxE experiments</h1>
<div id="individual-lines" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Individual lines</h2>
<p>GxE experiments were conducted on <code>G1_b2</code> (from ID representing drought tolerant genotype, but undergoing low summer heat stress) and <code>G2_b24</code> (from UT representing drought sensitive genotype, but undergoing high summer heat stress) individual lines. 30 plantlets were generated per individual line per treatment (n=30) with 5 plantlets for associated control treatments (n=5). Please see Figure <a href="#fig:timetab">3.1</a> for more details (see below for more details). Plantlets were grown in PhytoCon 946 ml vessels with XX gr of soil (4 sand: 1 vermiculite ratio) and acclimatized to condtions in the P4 Percival culture chamber. One week prior to imposing treatment a sealed plastic lid covering the soil was applied to prevent evaporation and only support water movement via transpiration. This setting was key to infer timing of stomatal closure as a response to applied treatments (see below for more details).</p>
</div>
<div id="vital-measurements" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Vital measurements</h2>
<p>Soil moisture, water potential (<span class="math inline">\(\Psi~soil~\)</span>) and air relative humidity were monitored during treatments to assess plantlets transpiration rates and evaluate stomatal closure using XXX devices. In the case of D and DH treatments, plantlets were harvested when stomata were fully closed 9meaning that they are entering into starvation to avoid death by cavitation; see below for more details). Leaf sizes were inadequate to measure stomatal conductance and we therefore had to apply a different approach to measure this process. This was estimated by comparing night vs. day soil moisture data using a t-test (here we were after H0 = no significant difference between the two datasets). Upon stomatal closure, transpiration would stop therefore soil moisture should remain identical between night and day. This assumption is only correct when evaporation is null and that variation in soil moisture is only driven by transpiration.</p>
<p>Estimating this process will be facilitated by the lid preventing evaporation (see above). To further estimate cavitation upon stomatal closure stem xylem pressure (<span class="math inline">\(\Psi~x~\)</span>) was measured when plantlets were harvested using the XXX device. This trait corresponds to the plant’s ability to undergo high xylem pressure caused by drought resulting in air bubbles entering the xylem and disrupting the water flow therefore causing cavitation. To avoid death by cavitation, plants evolved mechanisms to close their stomata, which are mostly regulated by ABA. The ability of a plant to sustain high xylem pressure and therefore minimize cavitation can be predicted by its vessels’ pore diameters. In this context, stem and root xylem anatomies of our individual lines were quantified and compared using SEM and light microscopy. Finally, transpiration rates have been shown to be associated to stomata density. Stomata density on both side of the blade were quantified and compared between individual lines using SEM.</p>
<p><strong>GxE experiments:</strong> Drought (D), Heat (H) and drought x heat (DH) treatments were conducted on 30 plantlets per individual line per treatment. Treatments were conducted in the P4 Percival culture chamber and all plantlets were distributed on one shelf to avoid any light or temperature effects. To further investigate any culture chamber effect, plantlets were equally subdivided into three groups (each including 10 plantlets) depending on their position on the shelf: right side, middle and left side. Estimation of transpiration rates for each group were inferred by randomly selecting two plantlets per group and tracking changes of water content through time using a METER EC-5 probe. Five control (C) plantlets associated to each treatment were harvested at 0 d (just before applying the treatment) to enable assessing treatment efficacy and identifying differentially expressed genes (DEGs) in leaves and roots using RNA-seq technique. Treatment were stopped and data harvested upon inference of stomatal closure at mid-day. This was inferred by conducting t-tests were night and day water content data were compared. No significant difference in water content should be inferred between night and day data since plantlets closed their stomata during the day to avoid death by cavitation and therefore entered into starvation. Prior to merging the water content data, statistical tests were conducted to ensure that there is no cluture chamber effect on plantlet’s transpiration. Finally, the level of cavitation upon stomata closure was inferred by recording stem xylem pressure and compared to control plantlets.</p>
<p><strong>Transcriptomics:</strong> Transcriptomic analyses were conducted on leaf and root tissues. Since all individuals are clones, RNA extractions were conducted on 5 randomly selected plantlets and pooled (by tissue and treatment) to conduct Illumina NovaSeq RNA-seq. Controls associated to each treatment (D, H, DH) were kept distinct (= distinct libraries) to allow for accurate comparisons and identifications of DEGs. In this context, a total of 12 Illumina libraries were built and sequenced on 2 Illumina NovaSeq runs (= 6 libraries per run).</p>
</div>
<div id="sampling" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Sampling</h2>
<p><a href="Individual_lines.html">Individual lines</a> that have been acclimatized to natural environmental conditions are used for these experiments. Every plantlet within an individual line will have a specific barcode to monitor its progress. Two individual lines will be used: <code>G1_b2</code> (representing drought tolerant genotype) and <code>G2_b24</code> (representing drought sensitive genotype). We will be conducting trial GxE experiments using <code>G2_b27</code> to make sure that all the settings are correct. Please see Figure <a href="#fig:timetab">3.1</a> for more details.</p>
</div>
<div id="timetable" class="section level2" number="3.4">
<h2><span class="header-section-number">3.4</span> Timetable</h2>
<p>The timetable associated to the GxE experiments and subsequent RNA sequencing (RNA-seq) and bioinformatic analyses (= transcriptome assemblies) required to annotate the sagebrush genome is presented in Figure <a href="#fig:timetab">3.1</a>.</p>
<p>The time allocated for each phase of the workflow (and representing the time associated to process an individual line) is as follows:</p>
<ul>
<li><span style="color: black;font-weight:bold">Ex planting</span>: 11 weeks. Note that the first two weeks and the final week of this phase are taking place in the culture chamber (P4). Please see the <a href="Individual_lines.html">Propagation</a> for more details.</li>
<li><span style="color: grey;font-weight:bold">GxE experiments</span>: 4 weeks. This time should allow for conducting the three GxE experiments (T<sub>H</sub>, T<sub>D</sub>, T<sub>DH</sub>) back to back on one individual line. This will take place in the culture chamber (P4).</li>
<li><span style="color: green;font-weight:bold">RNA-seq</span>: 10 weeks. Two weeks are dedicated to conduct RNA extractions at BSU and shipping samples to sequencing facility. We estimate that eight weeks will be required to conduct the sequencing.</li>
<li><span style="color: blue;font-weight:bold">Bioinformatics</span>: 20 weeks. Assembling transcriptomes, conducting comparative analyses to isolate adaptive loci and starting genome annotation will take 20 weeks.</li>
</ul>
<div class="figure"><span id="fig:timetab"></span>
<img src="Annotation_files/figure-html/timetab-1.png" alt="Timetable associated to GxE experiments and subsequent RNA sequencing and biorinformatic analyses used for annotate the sagebrush genome. Please see text for more details on individual lines." width="672" />
<p class="caption">
Figure 3.1: Timetable associated to GxE experiments and subsequent RNA sequencing and biorinformatic analyses used for annotate the sagebrush genome. Please see text for more details on individual lines.
</p>
</div>
<div id="climatic-growing-conditions" class="section level3" number="3.4.1">
<h3><span class="header-section-number">3.4.1</span> Climatic growing conditions</h3>
<p>During the ex planting protocol, plantlets are exposed to the following climatic conditions in:</p>
<ul>
<li>On the cart in the laboratory: 24<span>&amp;deg and average of 20.3% RH.</li>
<li>In the culture chamber (P4): 24<span>&amp;deg and average of 22% RH.</li>
</ul>
</div>
</div>
<div id="drought-treatment" class="section level2" number="3.5">
<h2><span class="header-section-number">3.5</span> Drought treatment</h2>
<p><img src="Annotation_files/figure-html/droughtplot-1.png" width="672" /></p>
<p><strong>Growing conditions:</strong> at constant 24<span>°</span>C.</p>
<p>**Vital measureme</p>
<p><strong>Treatments:</strong></p>
<p>Two treatments will be conducted to assess the effect of drought on gene expression patterns:</p>
<ul>
<li>T<sub>C</sub>: well-watered (control).</li>
<li>T<sub>D</sub>: drought.</li>
</ul>
<p><strong>Sampling</strong></p>
<p>Each T<sub>D</sub> plantlet will be sampled at the onset of drought (upon wilting), whereas T<sub>C</sub> plantlets will only be sampled at the end of the experiment (when all the T<sub>D</sub> plantlets are wilted). A response curve will be drawn to estimate phenotypic plasticity within each individual line. Hydraulic conductivity upon sampling will be measured for each plantlet.</p>
<div id="heat-experiment" class="section level4" number="3.5.0.1">
<h4><span class="header-section-number">3.5.0.1</span> Heat experiment</h4>
<p><strong>Growing conditions:</strong> at constant 40<span>°</span>C. Based on data from UT2.</p>
<p>42<span>°</span>C for most herbaceous plants:</p>
<p>Tissue collected at 0 h (= control) and 5 h (=heat treatment) for <em>Brachypodium distachyon</em>. [See] here(<a href="https://www.frontiersin.org/articles/10.3389/fpls.2016.02067/full" class="uri">https://www.frontiersin.org/articles/10.3389/fpls.2016.02067/full</a>)</p>
<p>Tissue collected after 2 days for <em>Brassica alboglabra</em>. See [here] (<a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-019-5652-y" class="uri">https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-019-5652-y</a>)</p>
<p>45<span>°</span>C for a shrub (Ziziphus) for 5 days with tissue sampling on 0d (control), 1d, 3d, 5d.</p>
<p>Plantlets will be well-watered before conducting the experiment.</p>
<p><strong>Treatments:</strong></p>
<p>One treatment will be conducted to assess the effect of heat on gene expression patterns:</p>
<ul>
<li>T<sub>H</sub>: heat treatment.</li>
</ul>
<p><strong>Sampling</strong></p>
<p>Each T<sub>H</sub> plantlet will be sampled at the onset of heat (upon chlorosis). A response curve will be drawn to estimate phenotypic plasticity within each individual line. Hydraulic conductivity upon sampling will be measured for each individual.</p>
</div>
<div id="heat-and-drought-experiment" class="section level4" number="3.5.0.2">
<h4><span class="header-section-number">3.5.0.2</span> Heat and drought experiment</h4>
<p><strong>Growing conditions:</strong> Start at 24<span>°</span>C and gradually increase to 40<span>°</span>C. Every day, we will increase the temperature by 4 <span>°</span>C and it will take 5 days to reach 40 <span>°</span>C (temperatures: 24, 28, 32, 36, 40).</p>
<p>Based on data from UT2.</p>
<p><strong>Treatments:</strong></p>
<p>Two treatments will be conducted to assess the effect of drought on gene expression patterns:</p>
<ul>
<li>T<sub>CH</sub>: well-watered (control), but experiencing gradual increase of heat.</li>
<li>T<sub>D+H</sub>: Drought and heat stress.</li>
</ul>
<p><strong>Sampling</strong></p>
<p>The onset of drought for each plantlet will be recorded (when they start wilting), but tissue will be sampled at chlorosis (= heat stress). We hypothesize that these two phenotypes might co-occur due to the growing conditions. T<sub>D+H</sub> individuals will be sampled when these show sign of chlorosis (= heat stress). Response curves (associated to drought and heat phenotypes) will be drawn to estimate phenotypic plasticity within each individual line. Hydraulic conductivity upon sampling will be measured for each individual.</p>
</div>
</div>
<div id="transcriptomic-analyses" class="section level2" number="3.6">
<h2><span class="header-section-number">3.6</span> Transcriptomic analyses</h2>
</div>
<div id="rna-extraction" class="section level2" number="3.7">
<h2><span class="header-section-number">3.7</span> RNA extraction</h2>
<ol style="list-style-type: decimal">
<li>Randomly select 5 plantlet per treatment per individual line to obtain representation of transcriptomic response at individual line.</li>
<li>Weigh 10 mg of both leaf and root tissues per randomly selected plantlet.</li>
<li>Pool leaf tissues per treatment per individual line in a tube (= ready for RNA extraction).</li>
<li>Pool root tissues per treatment per individual line in a tube (= ready for RNA extraction).</li>
<li>Conduct RNA extractions.</li>
<li>RNA-seq</li>
</ol>
</div>
</div>
<div id="references" class="section level1" number="4">
<h1><span class="header-section-number">4</span> References</h1>
<div id="refs">

</div>
</div>
<div id="appendix-1" class="section level1 appendix" number="5">
<h1><span class="header-section-number">5</span> Appendix 1</h1>
<p>Citations of all R packages used to generate this report.</p>
<p>[1] J. Allaire, Y. Xie, J. McPherson, et al. <em>rmarkdown: Dynamic
Documents for R</em>. R package version 2.6. 2020. &lt;URL:
<a href="https://github.com/rstudio/rmarkdown" class="uri">https://github.com/rstudio/rmarkdown</a>&gt;.</p>
<p>[2] C. Boettiger. <em>knitcitations: Citations for Knitr Markdown Files</em>.
R package version 1.0.10. 2019. &lt;URL:
<a href="https://github.com/cboettig/knitcitations" class="uri">https://github.com/cboettig/knitcitations</a>&gt;.</p>
<p>[3] R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2019. &lt;URL: <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>&gt;.</p>
<p>[4] H. Wickham and J. Bryan. <em>usethis: Automate Package and Project
Setup</em>. R package version 2.0.0. 2020. &lt;URL:
<a href="https://CRAN.R-project.org/package=usethis" class="uri">https://CRAN.R-project.org/package=usethis</a>&gt;.</p>
<p>[5] H. Wickham, J. Hester, and W. Chang. <em>devtools: Tools to Make
Developing R Packages Easier</em>. R package version 2.3.2. 2020. &lt;URL:
<a href="https://CRAN.R-project.org/package=devtools" class="uri">https://CRAN.R-project.org/package=devtools</a>&gt;.</p>
<p>[6] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. ISBN 978-1138700109. Boca Raton, Florida: Chapman and
Hall/CRC, 2016. &lt;URL: <a href="https://github.com/rstudio/bookdown" class="uri">https://github.com/rstudio/bookdown</a>&gt;.</p>
<p>[7] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. R package version 0.21. 2020. &lt;URL:
<a href="https://github.com/rstudio/bookdown" class="uri">https://github.com/rstudio/bookdown</a>&gt;.</p>
<p>[8] Y. Xie. <em>Dynamic Documents with R and knitr</em>. 2nd. ISBN
978-1498716963. Boca Raton, Florida: Chapman and Hall/CRC, 2015. &lt;URL:
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>&gt;.</p>
<p>[9] Y. Xie. <em>formatR: Format R Code Automatically</em>. R package version
1.7. 2019. &lt;URL: <a href="https://github.com/yihui/formatR" class="uri">https://github.com/yihui/formatR</a>&gt;.</p>
<p>[10] Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in
R.” In: <em>Implementing Reproducible Computational Research</em>. Ed. by V.
Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and
Hall/CRC, 2014. &lt;URL:
<a href="http://www.crcpress.com/product/isbn/9781466561595" class="uri">http://www.crcpress.com/product/isbn/9781466561595</a>&gt;.</p>
<p>[11] Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.30. 2020. &lt;URL:
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>&gt;.</p>
<p>[12] Y. Xie, J. Allaire, and G. Grolemund. <em>R Markdown: The Definitive
Guide</em>. ISBN 9781138359338. Boca Raton, Florida: Chapman and Hall/CRC,
2018. &lt;URL: <a href="https://bookdown.org/yihui/rmarkdown" class="uri">https://bookdown.org/yihui/rmarkdown</a>&gt;.</p>
<p>[13] Y. Xie, C. Dervieux, and E. Riederer. <em>R Markdown Cookbook</em>. ISBN
9780367563837. Boca Raton, Florida: Chapman and Hall/CRC, 2020. &lt;URL:
<a href="https://bookdown.org/yihui/rmarkdown-cookbook" class="uri">https://bookdown.org/yihui/rmarkdown-cookbook</a>&gt;.</p>
</div>
<div id="appendix-2" class="section level1 appendix" number="6">
<h1><span class="header-section-number">6</span> Appendix 2</h1>
<p>Version information about R, the operating system (OS) and attached or R loaded packages. This appendix was generated using <code>sessionInfo()</code>.</p>
<pre><code>## R version 3.6.1 (2019-07-05)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS Mojave 10.14.6
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] png_0.1-7                gargle_0.5.0             formattable_0.2.0.1     
##  [4] leaflet_2.0.3            googlesheets4_0.2.0      kableExtra_1.2.1        
##  [7] dplyr_1.0.2              kfigr_1.2                scales_1.1.1            
## [10] lubridate_1.7.9.2        MASS_7.3-53              forcats_0.5.0           
## [13] TreeTools_1.4.0          ggridges_0.5.2           stringr_1.4.0           
## [16] ape_5.4-1                ggtree_2.0.4             ggpubr_0.4.0            
## [19] ggplot2_3.3.3            chisq.posthoc.test_0.1.2 DT_0.16                 
## [22] lsmeans_2.30-0           emmeans_1.5.2-1          lmtest_0.9-38           
## [25] zoo_1.8-8                pscl_1.5.5               RColorBrewer_1.1-2      
## [28] gplots_3.1.0             devtools_2.3.2           usethis_2.0.0           
## [31] formatR_1.7              knitcitations_1.0.10     bookdown_0.21           
## [34] rmarkdown_2.6            knitr_1.30              
## 
## loaded via a namespace (and not attached):
##   [1] readxl_1.3.1        backports_1.2.1     fastmatch_1.1-0    
##   [4] plyr_1.8.6          igraph_1.2.6        lazyeval_0.2.2     
##   [7] crosstalk_1.1.0.1   digest_0.6.27       htmltools_0.5.0    
##  [10] fansi_0.4.1         magrittr_2.0.1      memoise_1.1.0      
##  [13] openxlsx_4.2.2      remotes_2.2.0       R.utils_2.10.1     
##  [16] prettyunits_1.1.1   colorspace_2.0-0    rvest_0.3.6        
##  [19] haven_2.3.1         rbibutils_1.4       xfun_0.20          
##  [22] callr_3.5.1         crayon_1.3.4        jsonlite_1.7.2     
##  [25] phangorn_2.5.5      glue_1.4.2          gtable_0.3.0       
##  [28] webshot_0.5.2       R.cache_0.14.0      car_3.0-10         
##  [31] pkgbuild_1.2.0      abind_1.4-5         mvtnorm_1.1-1      
##  [34] bibtex_0.4.2.3      rstatix_0.6.0       Rcpp_1.0.5         
##  [37] viridisLite_0.3.0   xtable_1.8-4        tidytree_0.3.3     
##  [40] foreign_0.8-75      bit_4.0.4           htmlwidgets_1.5.3  
##  [43] httr_1.4.2          ellipsis_0.3.1      pkgconfig_2.0.3    
##  [46] R.methodsS3_1.8.1   tidyselect_1.1.0    rlang_0.4.10       
##  [49] munsell_0.5.0       cellranger_1.1.0    tools_3.6.1        
##  [52] cli_2.2.0           generics_0.1.0      broom_0.7.1        
##  [55] evaluate_0.14       yaml_2.2.1          RefManageR_1.2.12  
##  [58] processx_3.4.5      bit64_4.0.5         fs_1.5.0           
##  [61] zip_2.1.1           caTools_1.18.0      purrr_0.3.4        
##  [64] nlme_3.1-149        R.oo_1.24.0         xml2_1.3.2         
##  [67] compiler_3.6.1      rstudioapi_0.13     curl_4.3           
##  [70] testthat_3.0.1      ggsignif_0.6.0      treeio_1.10.0      
##  [73] tibble_3.0.4        stringi_1.5.3       highr_0.8          
##  [76] ps_1.5.0            desc_1.2.0          lattice_0.20-41    
##  [79] Matrix_1.2-18       vctrs_0.3.6         pillar_1.4.7       
##  [82] lifecycle_0.2.0     BiocManager_1.30.10 Rdpack_2.1         
##  [85] estimability_1.3    data.table_1.13.6   bitops_1.0-6       
##  [88] gbRd_0.4-11         R6_2.5.0            KernSmooth_2.23-17 
##  [91] rio_0.5.16          codetools_0.2-16    sessioninfo_1.1.1  
##  [94] gtools_3.8.2        assertthat_0.2.1    pkgload_1.1.0      
##  [97] rprojroot_2.0.2     withr_2.3.0         parallel_3.6.1     
## [100] hms_0.5.3           quadprog_1.5-8      grid_3.6.1         
## [103] tidyr_1.1.2         coda_0.19-4         rvcheck_0.1.8      
## [106] carData_3.0-4       googledrive_1.0.1</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
