<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Biomass production</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.16/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<script src="site_libs/jszip-1.10.20/jszip.min.js"></script>
<script src="site_libs/pdfmake-1.10.20/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.10.20/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Sagebrush Genome Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Individual_lines.html">Propagation</a>
</li>
<li>
  <a href="Biomass_production.html">Biomass</a>
</li>
<li>
  <a href="Sequencing.html">Genome</a>
</li>
<li>
  <a href="Protocols.html">Protocols</a>
</li>
<li>
  <a href="Team.html">Team</a>
</li>
<li>
  <a href="funding.html">Acknowledgements</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Biomass production</h1>

</div>


<div id="biomass-requirements-for-genome-project" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Biomass requirements for genome project</h1>
<p>We aim at applying the newly developed <em>in vitro</em> tissue culture technique <span class="citation">(<a href="#ref-Barron2020" role="doc-biblioref">Barron et al., 2020</a>)</span> to the <code>G2_b24_1</code> individual line (from UTT2, drought-sensitive genotype; see Figure <a href="#fig:plantlets">1.1</a>) to produce the ca. 120 gr of leaf biomass necessary for genome sequencing, phasing and annotation. As mentioned in our introduction, we also need to make sure that we are maintaining the individual line beyond biomass production to support genome to phenome research (based on GxE experiments).</p>
<div class="figure" style="text-align: center"><span id="fig:plantlets"></span>
<img src="Figures/G2_b20_1_15wk_old_plantlet.jpg" alt="Example of a 15-week old plantlet grown in vitro in Magenta vessel." width="40%" />
<p class="caption">
Figure 1.1: Example of a 15-week old plantlet grown in vitro in Magenta vessel.
</p>
</div>
<div id="adapting-the-approach-to-produce-biomass" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Adapting the approach to produce biomass</h2>
<p>Our propagation method (see <a href="Individual_lines.html">here</a> for more details) to maintain individual line <em>in vitro</em> works very well, but after 8-10 weeks of growth a plantlet only generates ca. 0.15 gr of biomass for genome sequencing. This is only when allocating 1 plantlet per Magenta vessel. Indeed, sagebrush are competing using chemical compounds and putting several individuals per Magenta vessel decreases growth and increases mortality. Overall, if we were to apply to approach to generate the necessary 120 gr for the sagebrush genome project, we would need <strong>800 plantlets</strong>. This large number of plantlet is logistically challenging and could prove difficult to handle for our small to middle size lab. This is even more challenging because although our methodology has a high (&gt;90%) rooting response, the survival rate during the <span style="color: black;font-weight:bold">growth</span> step is highly variable between individual lines. For instance, the top performer as identified by <span class="citation"><a href="#ref-Barron2020" role="doc-biblioref">Barron et al.</a> (<a href="#ref-Barron2020" role="doc-biblioref">2020</a>)</span>, <code>G2_b27_1</code>, has only a 45% survival rate in the second round of propagation, whereas <code>G2_b24_1</code> has a 80% survival rate at the same stage. This evidence means that far more plantlets would have to be cultivated to reach the <strong>800 plantlets</strong> mark (corresponding to 120 gr.). This latter estimate is without accounting for a cushion to maintain the individual line for genome to phenome research. In this context, we have favored i) selecting an individual with a high survival rate and ii) growing plantlets for a longer period to obtain more biomass. To differentiate this stage of the propagation to our regular steps, we will be referring to the allocation of plantlets to biomass production as <span style="color: orange;font-weight:bold">biobanking</span> (see below for more details).</p>
<div id="using-15-week-old-g2_b24_1-plantlets-to-produce-biomass" class="section level3" number="1.1.1">
<h3><span class="header-section-number">1.1.1</span> Using 15-week old G2_b24_1 plantlets to produce biomass</h3>
<p>After 15 weeks of <em>in vitro</em> tissue culture, <strong>a plantlet produces ca. 0.8 gr. of leaf biomass</strong> that can be used for sequencing (Figure <a href="#fig:plantletweight">1.2</a>). Thus, <strong>a minimum of 150 plantlets have to be produced</strong> to sequence the genome (see Table <a href="#tab:tabproject"><strong>??</strong></a>). This number is much more manageable and means that all the plantlets can be grown in individual Magenta vessels under optimum growth conditions in our Percival culture chamber. When ready, the biomass will be flash frozen (using liquid nitrogen) and stored at -80C before being shipped to the sequencing facility. However, the estimation of biomass provided here does not account for a DNA extraction trial required to optimize protocol (we need at least 15<span>µ</span>g of HMW DNA for PacBio sequencing). We will have to discuss this with the project manager at Dovetail Genomics.</p>
<div class="figure" style="text-align: center"><span id="fig:plantletweight"></span>
<img src="Figures/G2_b20_1_Biomass_0.855g_after2min.jpg" alt="A 15-week old plantlet grown in vitro in Magenta vessel generates 0.8 gr of leaf biomass for sequencing." width="50%" />
<p class="caption">
Figure 1.2: A 15-week old plantlet grown in vitro in Magenta vessel generates 0.8 gr of leaf biomass for sequencing.
</p>
</div>
</div>
</div>
</div>
<div id="planning-biomass-production-for-sequencing" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Planning biomass production for sequencing</h1>
<p>To be able to plan the biomass production for genome sequencing, while also maintaining the individual line, we have developed an R function called <em>propagationPred</em>. This function incorporates 4 steps (<span style="color: black;font-weight:bold">Growth</span>, <span style="color: green;font-weight:bold">Cutting</span>, <span style="color: blue;font-weight:bold">Rooting</span> and <span style="color: orange;font-weight:bold">Biobanking</span>) as well as the rooting and survival rates. In addition, the use can define duration for the <span style="color: black;font-weight:bold">Growth</span>, <span style="color: blue;font-weight:bold">Rooting</span> and <span style="color: orange;font-weight:bold">Biobanking</span> steps, which allows establishing a schedule. Finally, this function also incorporates details on times necessary for preparing media and number of plates and Magenta vessels required to complete propagation protocols. The function is presented below.</p>
<div id="presenting-r-propagationpred-function" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Presenting R <em>propagationPred</em> function</h2>
<p><em>propagationPred</em> depends on the following set of arguments provided by the user:</p>
<ul>
<li><code>n</code>: number of plantlets to seed the experiment.</li>
<li><code>ns</code>: number of shoot tips cut per plantlet (after the growth period).</li>
<li><code>r</code>: vector of rooting rates (0 to 1) at each rooting phase.</li>
<li><code>s</code>: vector of survival rates (0 to 1) at each growth phase.</li>
<li><code>g</code>: number of generations (for <em>in vitro</em> propagation).</li>
<li><code>biobank</code>: vector with proportion (0 to 1) of plantlets biobanked per generation (at end of growth) (e.g: c(0,0,0.4,0)). This argument is important since it is equal to the number of plantlets set aside for biomass production.</li>
<li><code>date_user</code>: starting date of the experiment (mdy from <em>ludridate</em> or any date format).</li>
<li><code>growth_time_g</code>: vector with number of days/weeks allocated for growth per <code>g</code> (how long will it take for your plantlets to produce <code>ns</code>).</li>
<li><code>biobanking_time</code>: one value (either weeks() or days()) compatible with <em>lubridate</em> describing the duration of biobanking to generated the biomass (here 15 weeks corresponding to 0.8 gr).</li>
<li><code>rooting_time_g</code>: vector with number of days/weeks for rooting per <code>g</code>. The rooting time is a key factor, you want to allocate enough time for roots to develop and growth.</li>
<li><code>biomass_plantlet</code>: numerical value of biomass of a single plantlet (here 0.8 gr).</li>
</ul>
<p>The function outputs a <code>data.frame</code> with the following columns:</p>
<ul>
<li><code>Generation</code>: Generation ID.</li>
<li><code>Type</code>: Type of <em>in vitro</em> activity (<code>Growth</code>, <code>Cutting</code>, <code>Rooting</code>, <code>Biobanking</code>).</li>
<li><code>Date_start</code> and <code>Date_end</code>: Start and end date associated to <code>Type</code>.</li>
<li><code>N_plant_start</code> and <code>N_plant_end</code>: Number of plantelts at stat and end dates.</li>
<li><code>N_vessels</code>: Number of plates (for <code>Rooting</code>) or Magenta (for <code>Growth</code>) vessels. 1 plate = 9 shoot tips, whereas rooted shoot tips are on individual Magenta vessels.<br />
</li>
<li><code>Volume_media_litre</code>: Amount of <code>Rooting</code> (0.05 liter per plate) or <code>Growth</code> (0.1 liter per Magenta vessel) media to be prepared.<br />
</li>
<li><code>N_Autoclave</code> and <code>Time_Autoclave (hrs)</code>: Number of autoclaves used to prepare <code>Rooting</code> (1 liter per autoclave) or <code>Growth</code> (60 Magenta vessels per autoclave) media and associated times (120 minutes for <code>Rooting</code> media vs. 90 minutes for <code>Growth</code> media).</li>
<li><code>Media_prep_time (hrs)</code> and <code>Time_Cutting (hrs)</code>: Lab our time to prepare <code>Rooting</code> (90 minutes for 1 liter) or <code>Growth</code> (1 hr for 60 boxes) media and conduct cutting (5 plantlets per hour).<br />
</li>
<li><code>Total_Biomass (gr)</code>: This is estimated based on fraction of plantlets allocated to biobanking (based on <code>biomass_plantlet</code>).</li>
</ul>
<p>The output can directly be used to plan <em>in vitro</em> propagation, but can also be easily used as input to produce a plot showing the different steps through time (see below).</p>
</div>
</div>
<div id="scheduling-propagation-of-g2_b24_1" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Scheduling propagation of G2_b24_1</h1>
<p>As mentioned above, we have selected the <code>G2_b24_1</code> individual line as candidate for the Sagebrush Genome Project. In this section, we are using <em>propagationPred</em> to:</p>
<ul>
<li>Provide timetable of biomass production, especially to identify when the biomass will be ready to be delivered to Dovetail Genomics for sequencing.</li>
<li>Logistically organize the biomass production.</li>
</ul>
<div id="settings-applied-for-analyses" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Settings applied for analyses</h2>
<p>The <em>propagationPred</em> function has been applied with the following parameters (see section <a href="#predcode">3.4</a> for R code):</p>
<ul>
<li><code>n</code>: 6 plantlets at generation 1.</li>
<li><code>ns</code>: 9.5 (on average) shoot tips per plantlets after 10 weeks of growth.</li>
<li><code>r</code>: 0.93 rooting rate (same at each generation).</li>
<li><code>s</code>: 0.8 survival rate (same at each generation).</li>
<li><code>g</code>: 4 generations of propagation.</li>
<li><code>biobank</code>: c(0,0,0.9,0). Meaning that 90% of plantlets will be biobanked at generation 3. Based on preliminary analyses, this should provide sufficient number of plantlets for biomass production.</li>
<li><code>date_user</code>: 2020-09-04 is the beginning of growth period at generation 1.</li>
<li><code>growth_time_g</code>: c(weeks(10),weeks(8),weeks(8),weeks(8)).</li>
<li><code>biobanking_time</code>: weeks(15). This corresponds to the duration required for a plantlet to produce 0.8 gr of leaf biomass.</li>
<li><code>rooting_time_g</code>: c(weeks(4),weeks(3),weeks(3),weeks(3)).</li>
<li><code>biomass_plantlet</code>: 0.8 (in gr).</li>
</ul>
</div>
<div id="timetable-of-biomass-production" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Timetable of biomass production</h2>
<p>A detailed timetable for each steps of the propagation procedure of the <code>G2_b24_1</code> individual line is available below.</p>
<div id="htmlwidget-e31c6757625d27cbad90" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e31c6757625d27cbad90">{"x":{"filter":"none","extensions":["Buttons"],"data":[["1","3","4","5","7","8","9","10","11","12","13"],["1","1","1","2","2","2","3","3","3","3","4"],["Growth","Cutting","Rooting","Growth","Cutting","Rooting","Growth","Biobanking","Cutting","Rooting","Growth"],["2020-09-04","2020-11-13","2020-11-13","2020-12-11","2021-02-05","2021-02-05","2021-02-26","2021-02-26","2021-04-23","2021-04-23","2021-05-14"],["2020-11-13","2020-11-13","2020-12-11","2021-02-05","2021-02-05","2021-02-26","2021-04-23","2021-06-11","2021-04-23","2021-05-14","2021-07-09"],["6","6","57","53","42","399","38","333","30","285","265"],["6","57","53","42","399","371","30","266","285","265","212"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Generation<\/th>\n      <th>Type<\/th>\n      <th>Date_start<\/th>\n      <th>Date_end<\/th>\n      <th>N_plant_start<\/th>\n      <th>N_plant_end<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf","print"],"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
<div id="key-highlights" class="section level3" number="3.2.1">
<h3><span class="header-section-number">3.2.1</span> Key highlights</h3>
<ul>
<li>On 2021-02-26, 333 rooted plantlets transferred into individual Magenta vessels will be <span style="color: orange;font-weight:bold">biobanked</span> in a dedicated Percival. Based on our prediction, <strong>266 plantlets will survived and produce 212.8 gr. of biomass by 2021-06-11</strong>.</li>
<li>The biomass production should largely exceed the needs of the project, which might suggest that we could deliver the biomass sooner. However, this remains a prediction, which will have to be confirmed.<br />
</li>
<li>To better visualize the timetable, a plot was produced and it highlights the key dates and facts (see Figure <a href="#fig:plotpred">3.1</a>).</li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:plotpred"></span>
<img src="Biomass_production_files/figure-html/plotpred-1.png" alt="Plot showing the time frame to conduct the in vitro propagation of G2_b24_1 to produce the biomass for the Sagebrush Genome Project and maintain the individual line." width="960" />
<p class="caption">
Figure 3.1: Plot showing the time frame to conduct the in vitro propagation of G2_b24_1 to produce the biomass for the Sagebrush Genome Project and maintain the individual line.
</p>
</div>
</div>
</div>
<div id="logistical-organization" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Logistical organization</h2>
<p>To prepare with each step of the propagation, we are providing key knowledge in the table below:</p>
<div id="htmlwidget-bdc1a8eb30daaaa6e162" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-bdc1a8eb30daaaa6e162">{"x":{"filter":"none","extensions":["Buttons"],"data":[["1","3","4","5","7","8","9","10","11","12","13"],["1","1","1","2","2","2","3","3","3","3","4"],["Growth","Cutting","Rooting","Growth","Cutting","Rooting","Growth","Biobanking","Cutting","Rooting","Growth"],["2020-09-04","2020-11-13","2020-11-13","2020-12-11","2021-02-05","2021-02-05","2021-02-26","2021-02-26","2021-04-23","2021-04-23","2021-05-14"],["2020-11-13","2020-11-13","2020-12-11","2021-02-05","2021-02-05","2021-02-26","2021-04-23","2021-06-11","2021-04-23","2021-05-14","2021-07-09"],["6",null,"7","53",null,"45","38","333",null,"32","265"],["1",null,"1","6",null,"3","4","1",null,"2","27"],["1",null,"1","1",null,"3","1","6",null,"2","5"],["1.5",null,"2","1.5",null,"6","1.5","9",null,"4","7.5"],["1",null,"1.5","1",null,"4.5","1","6",null,"3","5"],[null,"1.2",null,null,"8.4",null,null,null,"6",null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Generation<\/th>\n      <th>Type<\/th>\n      <th>Date_start<\/th>\n      <th>Date_end<\/th>\n      <th>N_vessels<\/th>\n      <th>Volume_media_litre<\/th>\n      <th>N_Autoclave<\/th>\n      <th>Time_Autoclave (hrs)<\/th>\n      <th>Media_prep_time (hrs)<\/th>\n      <th>Time_Cutting (hrs)<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf","print"],"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
<div id="key-highlights-1" class="section level3" number="3.3.1">
<h3><span class="header-section-number">3.3.1</span> Key highlights</h3>
<ul>
<li>The <span style="color: green;font-weight:bold">Cutting</span> on 2021-02-05 will require NA hours of labor. We will have to make sure to coordinate and take 2-3 days to complete this step.</li>
<li>A total of 399 Magenta vessels will be needed by 2021-02-05.</li>
</ul>
</div>
</div>
<div id="predcode" class="section level2" number="3.4">
<h2><span class="header-section-number">3.4</span> R code</h2>
<pre class="r"><code>#Source user-made function
source(&quot;Functions/Propagation.R&quot;)

#Apply propagationPred to top performer (G2_b24_1 from UTT2)
OUTdat &lt;- propagationPred(n=6, ns=9.5, r=rep(0.93,4), s=rep(0.8,4), g=4, biobank=c(0,0,0.9,0), date_user=mdy(&quot;11/13/2020&quot;)-weeks(10), growth_time_g=c(weeks(10),weeks(8),weeks(8),weeks(8)), biobanking_time = weeks(15), rooting_time_g=c(weeks(4),weeks(3),weeks(3),weeks(3)), biomass_plantlet = 0.8)

#Subset OUTdat to only contain rows of interests
OUTdat &lt;- OUTdat[1:11,]

#Plot table
DT::datatable(OUTdat, extensions = &#39;Buttons&#39;, options = list(dom = &#39;Blfrtip&#39;, buttons = c(&#39;copy&#39;, &#39;csv&#39;, &#39;excel&#39;, &#39;pdf&#39;, &#39;print&#39;)))

###~~~
#Plot data
###~~~
#pdf(&quot;Schedule_sagebrush_propagation_G2_b24_1.pdf&quot;)
#Populate plot
Type &lt;- levels(OUTdat$Type)[c(3,2,4,1)]
colSeg &lt;- c(&quot;black&quot;,&quot;green&quot;,&quot;blue&quot;,&quot;orange&quot;)

#Create empty plot
plot(x=c(as.Date(OUTdat$Date_start)), y= c(as.numeric(as.vector(OUTdat$N_plant_start))), 
     xlim= c(min(as.Date(OUTdat$Date_start)), max(as.Date(OUTdat$Date_end))+20), 
     ylim=c(0, max(as.numeric(as.vector(OUTdat$N_plant_start)))+50), type=&#39;n&#39;, xlab=&quot;Time (days)&quot;, ylab = &quot;Number of Plantlets&quot;) 

biomassTot &lt;- 120
biomass_plantlet &lt;- 0.8
#Add objective of N plantlets
abline(h=biomassTot/biomass_plantlet, col=&#39;grey&#39;)
text(x=as.Date(OUTdat$Date_start[1]), y=biomassTot/biomass_plantlet+10, labels=paste(&quot;Number of Plantlets for Project (N: &quot;, biomassTot/biomass_plantlet, &quot; = &quot;, biomassTot, &quot; gr. )&quot;, sep=&quot;&quot;), adj=0, col=&quot;grey&quot;, cex=.65)

#Add segments
for(i in 1:length(Type)){
  tmp &lt;- OUTdat[which(OUTdat$Type == Type[i]),]
  segments(x0=as.Date(tmp$Date_start), x1=as.Date(tmp$Date_end), y0=as.numeric(as.vector(tmp$N_plant_start)), y1=as.numeric(as.vector(tmp$N_plant_end)), col=colSeg[i], lwd=1.5)
  #Add points to better see beginning and end of phases
  points(x=as.Date(tmp$Date_start), y=as.numeric(as.vector(tmp$N_plant_start)), col=colSeg[i], pch=16, cex=.8)
  points(x=as.Date(tmp$Date_end), y=as.numeric(as.vector(tmp$N_plant_end)), col=colSeg[i], pch=16, cex=.8)
  
  if(Type[i] == &quot;Cutting&quot;){
    text(x=as.Date(tmp$Date_start), y=as.numeric(as.vector(tmp$N_plant_end))+20, labels = as.Date(tmp$Date_start), srt=45, adj=0.5, cex=.6)
  }
  if(Type[i] == &quot;Rooting&quot;){
    text(x=as.Date(tmp$Date_end), y=as.numeric(as.vector(tmp$N_plant_end))+8, labels = as.Date(tmp$Date_end), srt=45, adj=0, cex=.6)
  }
  if(Type[i] == &quot;Growth&quot;){
    #Start of experiment
    startExp &lt;- which(tmp$Date_start == as.character(min(as.Date(tmp$Date_start))))
    text(x=as.Date(tmp$Date_start)[startExp], y=as.numeric(as.vector(tmp$N_plant_start))[startExp]+8, labels = as.Date(tmp$Date_start)[startExp], srt=90, adj=0, cex=.6)
    
    #End of experiment
    endExp &lt;- which(tmp$Date_end == as.character(max(as.Date(tmp$Date_end))))
    text(x=as.Date(tmp$Date_end)[endExp], y=as.numeric(as.vector(tmp$N_plant_end))[endExp]+8, labels = as.Date(tmp$Date_end)[endExp], srt=90, adj=0, cex=.6)
  }
  if(Type[i] == &quot;Biobanking&quot;){
    #Add segment(s) showing when rooted shoot tips are split between growth and biobanking
    ystart &lt;- as.numeric(as.vector(OUTdat$N_plant_start[which(OUTdat$Date_start == tmp$Date_start &amp; OUTdat$Type == &quot;Growth&quot;)]))
    yend &lt;- as.numeric(as.vector(OUTdat$N_plant_end[which(OUTdat$Date_end == as.character(tmp$Date_start) &amp; OUTdat$Type == &quot;Rooting&quot;)]))
    segments(x0=as.Date(tmp$Date_start), x1=as.Date(tmp$Date_start), y0=ystart, y1=yend, lty=2) 
    text(x=as.Date(tmp$Date_start)-4, y=ystart, paste(&quot;Split rooted shoot tips:&quot;, paste(as.character(tmp$Type), &quot; (&quot;, tmp$N_plant_start, &quot;)&quot;, &quot; &amp; Growth (&quot;, ystart, &quot;)&quot;, sep=&quot;&quot;)), adj=0, srt=90, cex=.6)
    
    #End of experiment
    endExp &lt;- which(tmp$Date_end == as.character(max(as.Date(tmp$Date_end))))
    text(x=as.Date(tmp$Date_end)[endExp], y=as.numeric(as.vector(tmp$N_plant_end))[endExp]+8, labels = paste(as.Date(tmp$Date_end)[endExp], &quot; (&quot;, as.character(tmp$`Total_Biomass (gr)`), &quot; gr.)&quot;, sep=&#39;&#39;), srt=90, adj=0, cex=.6)
  }
  
}

#Add legend
legend(&quot;topleft&quot;, legend = Type, lty=1, col=colSeg, cex=.6)
#dev.off()</code></pre>
</div>
</div>
<div id="references" class="section level1" number="4">
<h1><span class="header-section-number">4</span> References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Barron2020" class="csl-entry">
<span class="smallcaps">Barron, R.</span>, <span class="smallcaps">P. Martinez</span>, <span class="smallcaps">M. Serpe</span>, and <span class="smallcaps">S. Buerki</span>. 2020. Development of an in vitro method of propagation for artemisia tridentata subsp. Tridentata to support genome sequencing and genotype-by-environment research. <em>Plants</em> 9: Available at: <a href="https://www.mdpi.com/2223-7747/9/12/1717">https://www.mdpi.com/2223-7747/9/12/1717</a>.
</div>
</div>
</div>
<div id="appendix-1" class="section level1 appendix" number="5">
<h1><span class="header-section-number">5</span> Appendix 1</h1>
<p>Citations of all R packages used to generate this report.</p>
<p>[1] J. Allaire, Y. Xie, J. McPherson, et al. <em>rmarkdown: Dynamic
Documents for R</em>. R package version 2.6. 2020. &lt;URL:
<a href="https://github.com/rstudio/rmarkdown" class="uri">https://github.com/rstudio/rmarkdown</a>&gt;.</p>
<p>[2] C. Boettiger. <em>knitcitations: Citations for Knitr Markdown Files</em>.
R package version 1.0.10. 2019. &lt;URL:
<a href="https://github.com/cboettig/knitcitations" class="uri">https://github.com/cboettig/knitcitations</a>&gt;.</p>
<p>[3] G. Grolemund and H. Wickham. “Dates and Times Made Easy with
lubridate.” In: <em>Journal of Statistical Software</em> 40.3 (2011), pp.
1-25. &lt;URL: <a href="https://www.jstatsoft.org/v40/i03/" class="uri">https://www.jstatsoft.org/v40/i03/</a>&gt;.</p>
<p>[4] R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2019. &lt;URL: <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>&gt;.</p>
<p>[5] V. Spinu, G. Grolemund, and H. Wickham. <em>lubridate: Make Dealing
with Dates a Little Easier</em>. R package version 1.7.9.2. 2020. &lt;URL:
<a href="https://CRAN.R-project.org/package=lubridate" class="uri">https://CRAN.R-project.org/package=lubridate</a>&gt;.</p>
<p>[6] H. Wickham and J. Bryan. <em>usethis: Automate Package and Project
Setup</em>. R package version 2.0.0. 2020. &lt;URL:
<a href="https://CRAN.R-project.org/package=usethis" class="uri">https://CRAN.R-project.org/package=usethis</a>&gt;.</p>
<p>[7] H. Wickham, J. Hester, and W. Chang. <em>devtools: Tools to Make
Developing R Packages Easier</em>. R package version 2.3.2. 2020. &lt;URL:
<a href="https://CRAN.R-project.org/package=devtools" class="uri">https://CRAN.R-project.org/package=devtools</a>&gt;.</p>
<p>[8] H. Wickham and D. Seidel. <em>scales: Scale Functions for
Visualization</em>. R package version 1.1.1. 2020. &lt;URL:
<a href="https://CRAN.R-project.org/package=scales" class="uri">https://CRAN.R-project.org/package=scales</a>&gt;.</p>
<p>[9] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. ISBN 978-1138700109. Boca Raton, Florida: Chapman and
Hall/CRC, 2016. &lt;URL: <a href="https://github.com/rstudio/bookdown" class="uri">https://github.com/rstudio/bookdown</a>&gt;.</p>
<p>[10] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. R package version 0.21. 2020. &lt;URL:
<a href="https://github.com/rstudio/bookdown" class="uri">https://github.com/rstudio/bookdown</a>&gt;.</p>
<p>[11] Y. Xie. <em>Dynamic Documents with R and knitr</em>. 2nd. ISBN
978-1498716963. Boca Raton, Florida: Chapman and Hall/CRC, 2015. &lt;URL:
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>&gt;.</p>
<p>[12] Y. Xie. <em>formatR: Format R Code Automatically</em>. R package version
1.7. 2019. &lt;URL: <a href="https://github.com/yihui/formatR" class="uri">https://github.com/yihui/formatR</a>&gt;.</p>
<p>[13] Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in
R.” In: <em>Implementing Reproducible Computational Research</em>. Ed. by V.
Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and
Hall/CRC, 2014. &lt;URL:
<a href="http://www.crcpress.com/product/isbn/9781466561595" class="uri">http://www.crcpress.com/product/isbn/9781466561595</a>&gt;.</p>
<p>[14] Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.30. 2020. &lt;URL:
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>&gt;.</p>
<p>[15] Y. Xie, J. Allaire, and G. Grolemund. <em>R Markdown: The Definitive
Guide</em>. ISBN 9781138359338. Boca Raton, Florida: Chapman and Hall/CRC,
2018. &lt;URL: <a href="https://bookdown.org/yihui/rmarkdown" class="uri">https://bookdown.org/yihui/rmarkdown</a>&gt;.</p>
<p>[16] Y. Xie, C. Dervieux, and E. Riederer. <em>R Markdown Cookbook</em>. ISBN
9780367563837. Boca Raton, Florida: Chapman and Hall/CRC, 2020. &lt;URL:
<a href="https://bookdown.org/yihui/rmarkdown-cookbook" class="uri">https://bookdown.org/yihui/rmarkdown-cookbook</a>&gt;.</p>
</div>
<div id="appendix-2" class="section level1 appendix" number="6">
<h1><span class="header-section-number">6</span> Appendix 2</h1>
<p>Version information about R, the operating system (OS) and attached or R loaded packages. This appendix was generated using <code>sessionInfo()</code>.</p>
<pre><code>## R version 3.6.1 (2019-07-05)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS Mojave 10.14.6
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] formattable_0.2.0.1      leaflet_2.0.3            googlesheets4_0.2.0     
##  [4] kableExtra_1.2.1         dplyr_1.0.2              kfigr_1.2               
##  [7] MASS_7.3-53              forcats_0.5.0            TreeTools_1.4.0         
## [10] ggridges_0.5.2           stringr_1.4.0            ape_5.4-1               
## [13] ggtree_2.0.4             ggpubr_0.4.0             ggplot2_3.3.3           
## [16] chisq.posthoc.test_0.1.2 DT_0.16                  lsmeans_2.30-0          
## [19] emmeans_1.5.2-1          lmtest_0.9-38            zoo_1.8-8               
## [22] pscl_1.5.5               RColorBrewer_1.1-2       gplots_3.1.0            
## [25] scales_1.1.1             lubridate_1.7.9.2        devtools_2.3.2          
## [28] usethis_2.0.0            formatR_1.7              knitcitations_1.0.10    
## [31] bookdown_0.21            rmarkdown_2.6            knitr_1.30              
## 
## loaded via a namespace (and not attached):
##   [1] readxl_1.3.1        backports_1.2.1     fastmatch_1.1-0    
##   [4] plyr_1.8.6          igraph_1.2.6        lazyeval_0.2.2     
##   [7] crosstalk_1.1.0.1   digest_0.6.27       htmltools_0.5.0    
##  [10] fansi_0.4.1         magrittr_2.0.1      memoise_1.1.0      
##  [13] openxlsx_4.2.2      remotes_2.2.0       R.utils_2.10.1     
##  [16] prettyunits_1.1.1   colorspace_2.0-0    rvest_0.3.6        
##  [19] haven_2.3.1         rbibutils_1.4       xfun_0.20          
##  [22] callr_3.5.1         crayon_1.3.4        jsonlite_1.7.2     
##  [25] phangorn_2.5.5      glue_1.4.2          gtable_0.3.0       
##  [28] webshot_0.5.2       R.cache_0.14.0      car_3.0-10         
##  [31] pkgbuild_1.2.0      abind_1.4-5         mvtnorm_1.1-1      
##  [34] bibtex_0.4.2.3      rstatix_0.6.0       Rcpp_1.0.5         
##  [37] viridisLite_0.3.0   xtable_1.8-4        tidytree_0.3.3     
##  [40] foreign_0.8-75      bit_4.0.4           htmlwidgets_1.5.3  
##  [43] httr_1.4.2          ellipsis_0.3.1      pkgconfig_2.0.3    
##  [46] R.methodsS3_1.8.1   tidyselect_1.1.0    rlang_0.4.10       
##  [49] munsell_0.5.0       cellranger_1.1.0    tools_3.6.1        
##  [52] cli_2.2.0           generics_0.1.0      broom_0.7.1        
##  [55] evaluate_0.14       yaml_2.2.1          RefManageR_1.2.12  
##  [58] processx_3.4.5      bit64_4.0.5         fs_1.5.0           
##  [61] zip_2.1.1           caTools_1.18.0      purrr_0.3.4        
##  [64] nlme_3.1-149        R.oo_1.24.0         xml2_1.3.2         
##  [67] compiler_3.6.1      rstudioapi_0.13     curl_4.3           
##  [70] testthat_3.0.1      ggsignif_0.6.0      treeio_1.10.0      
##  [73] tibble_3.0.4        stringi_1.5.3       highr_0.8          
##  [76] ps_1.5.0            desc_1.2.0          lattice_0.20-41    
##  [79] Matrix_1.2-18       vctrs_0.3.6         pillar_1.4.7       
##  [82] lifecycle_0.2.0     BiocManager_1.30.10 Rdpack_2.1         
##  [85] estimability_1.3    data.table_1.13.6   bitops_1.0-6       
##  [88] gbRd_0.4-11         R6_2.5.0            KernSmooth_2.23-17 
##  [91] rio_0.5.16          codetools_0.2-16    sessioninfo_1.1.1  
##  [94] gtools_3.8.2        assertthat_0.2.1    pkgload_1.1.0      
##  [97] rprojroot_2.0.2     withr_2.3.0         parallel_3.6.1     
## [100] hms_0.5.3           quadprog_1.5-8      grid_3.6.1         
## [103] tidyr_1.1.2         coda_0.19-4         rvcheck_0.1.8      
## [106] carData_3.0-4       googledrive_1.0.1</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
