--- 
title: 'Sagebrush Biomass Production'
output:
  bookdown::html_document2: 
    toc: TRUE
    toc_depth: 4
    toc_float: TRUE
link-citations: yes
fontsize: 12pt
bibliography: References.bib
---

```{r packages, echo=FALSE, warning=FALSE, include=FALSE}
library(knitr)
library(rmarkdown)
library(bookdown)
#library(distill)
library(knitcitations)
library(formatR)
library(devtools)
library(usethis)
library(lubridate)
library(scales)

#Generate BibTex citation file for all R packages used to produce report
knitr::write_bib(.packages(), file = 'packages.bib')
```


```{r setup, include=FALSE, cache=FALSE, message = FALSE}
#Load packages
library(knitr)
library(rmarkdown)
# Chunk options: see http://yihui.name/knitr/options/ ###

## Text results
opts_chunk$set(echo = TRUE, warning = TRUE, message = TRUE, include = TRUE)

## Code decoration
#opts_chunk$set(tidy = TRUE, R.options = list(width = 60), highlight = TRUE)

## Caching code
opts_chunk$set(cache = 2, cache.path = "cache/")

## Plots
#opts_chunk$set(fig.path = "Figures_MS/", dev=c('pdf', 'png'), dpi = 300)

## Locate figures as close as possible to requested position (=code)
#opts_chunk$set(fig.pos = "H")
```

# Biomass requirements for genome project

We aim at applying the newly developed *in vitro* tissue culture technique [@Barron2020] to the G2_b24_1 individual line (from UTT2, drought-sensitive genotype; see Figure \@ref(fig:plantlets)) to produce the ca. 120 gr of leaf biomass necessary for genome sequencing, phasing and annotation. As mentioned in our introduction, we also need to make sure that we are maintaining the individual line beyond biomass production. 

```{r plantlets, echo=F, eval=T, out.width="50%", fig.align='center', fig.cap= "Example of a 15-week old plantlet grown in vitro in Magenta vessel."}
knitr::include_graphics("Figures/G2_b20_1_15wk_old_plantlet.jpg")
```

# Overview of *in vitro* tissue culture

The *in vitro* tissue culture propagation that we have recently published for *Artemisia tridentata* subsp. *tridentata* is composed of three steps [@Barron2020]:

1. <span style="color: black;font-weight:bold">Growth</span>: During the first generation, we started from seeds, but later this step was based on rooted shoot tips cultured on MS growth media in Magenta vessels.
2. <span style="color: green;font-weight:bold">Cutting</span>: Here, shoot tips are cut from the grown seedlings or plantlets and transferred into rooting media.
3. <span style="color: blue;font-weight:bold">Rooting</span>: Shoot tips are cultivated on a rooting media containing IBA growth regulator (1 ml/l) in plates to initiate rooting. Rooted shoot tips are then transferred into growth media and we are back to step 1.

The <span style="color: black;font-weight:bold">Growth</span> and <span style="color: blue;font-weight:bold">Rooting</span> steps are taking 8 to 10 and 3 weeks respectively, while <span style="color: green;font-weight:bold">Cutting</span> is labor intensive and usually takes 1 hour to process 5 plantlets. After 8-10 weeks of growth a plantlet generates ca. 9 shoot tips, which are transferred into 1 plate of rooting media to initiate rooting (after 3 weeks of growth). This process works very well to keep the line, but at after 8-10 weeks of growth a plantlet only generate ca. 0.15 gr of biomass for genome sequencing. If we were to apply to approach to generate the necessary 120 gr, we would need `r 120/0.15` plantlets. This large number of plantlet is logistically challenging and could prove difficult to handle for our small lab. This is even more challenging because although our methodology has a high (>90%) rooting response, the survival rate during the <span style="color: black;font-weight:bold">Growth</span> step is highly variable between individual lines between 45 and 80%. This means that far more plantlets will have to be grown to reach the `r 120/0.15` plantlet mark and this is without accounting for a cushion to be able to maintain the line. In this context, we have favored i) selecting an individual with a high survival rate and ii) growing plantlets for a longer period to obtain more biomass.    

## How many plantlets of G2_b24_1 do we need to generate the required biomass?

After 15 weeks of *in vitro* tissue culture, **a plantlet produces 0.8 gr. of leaf biomass** that can be used for the genome project (Figure \@ref(fig:plantletweight)). To meet our requirements **a minimum of `r 120/0.8` plantlets have to be produced at BSU** (see Table \@ref(tab:tabproject)). 

The biomass can be flash frozen (using liquid nitrogen) and stored at -80C. The estimation of biomass provided here does not account for a DNA extraction trial required to optimize protocol (we need at least 15ug of HMW DNA for PacBio sequencing). We will have to discuss with the project manager at Dovetail Genomics.

```{r plantletweight, echo=F, eval=T, out.width="50%", fig.align='center', fig.cap= "A 15-week old plantlet grown in vitro in Magenta vessel generates 0.8 gr of leaf biomass for sequencing."}
knitr::include_graphics("Figures/G2_b20_1_Biomass_0.855g_after2min.jpg")
```



## Presenting the *propagationPred* R function

This function depends on the following arguments provided by the user:

- `n`: number of plantlets to seed the experiment.
- `ns`: number of shoot tips cut per plantlet (after the growth period).
- `r`: vector of rooting rates (0 to 1) at each rooting phase. 
- `s`: vector of survival rates (0 to 1) at each growth phase.
- `g`: number of generations (for *in vitro* propagation).
- `biobank`: vector with proportion (0 to 1) of plantlets biobanked per generation (at end of growth) (e.g: c(0,0,0.4,0)). This argument is important since it is equal to the number of plantlets set aside for biomass production.
- `date_user`: starting date of the experiment (mdy from *ludridate* or any date format).
- `growth_time_g`: vector with number of days/weeks allocated for growth per `g` (how long will it take for your plantlets to produce `ns`).
- `biobanking_time`: one value (either weeks() or days()) compatible with *lubridate* describing the duration of biobanking to generated the biomass (here 15 weeks corresponding to 0.8 gr).
- `rooting_time_g`: vector with number of days/weeks for rooting per `g`. The rooting time is a key factor, you want to allocate enough time for roots to develop and growth.
- `biomass_plantlet`: numerical value of biomass of a single plantlet (here 0.8 gr).

The function outputs a `data.frame` with the following columns:

- `Generation`: Generation ID.
- `Type`: Type of *in vitro* activity (`Growth`, `Cutting`, `Rooting`, `Biobanking`).
- `Date_start` and `Date_end`: Start and end date associated to `Type`.
- `N_plant_start` and `N_plant_end`: Number of plantelts at stat and end dates.
- `N_vessels`: Number of plates (for `Rooting`) or Magenta (for `Growth`) vessels. 1 plate = 9 shoot tips, whereas rooted shoot tips are on individual Magenta vessels.            
- `Volume_media_litre`: Amount of `Rooting` (0.05 liter per plate) or `Growth` (0.1 liter per Magenta vessel) media to be prepared.    
- `N_Autoclave` and `Time_Autoclave (hrs)`: Number of autoclaves used to prepare `Rooting` (1 liter per autoclave) or `Growth` (60 Magenta vessels per autoclave) media and associated times (120 minutes for `Rooting` media vs. 90 minutes for `Growth` media). 
- `Media_prep_time (hrs)` and `Time_Cutting (hrs)`: Lab our time to prepare `Rooting` (90 minutes for 1 liter) or `Growth` (1 hr for 60 boxes) media and conduct cutting (5 plantlets per hour).   
- `Total_Biomass (gr)`: This is estimated based on fraction of plantlets allocated to biobanking (based on `biomass_plantlet`).    




## Biomass requirements per sequencing technology

Table \@ref(tab:tabproject) provides a summary of sequencing technologies applied in this project (see [Sequencing & Bioinformatics](Sequencing.html) for more details), their purpose and biomass requirements. As mentioned above, these data do not account for a preliminary DNA extraction trial.

```{r tabproject, echo=FALSE, eval=TRUE}
#Table with data on sequencing and number of plantlets requiered for project
dat <- data.frame(Type = c("PacBio sequencing", "Proximity ligation (Hi-C for phasing genome)", "RNA-seq"),
                    #Add purpose
                    Purpose = c("De novo genome assembly", "Phasing", "Annotation"),
                    # add requirements 
                    `Unit/Biomass` = c("1 cell = 20 gr", "1 Illumina library = 6 gr", "1 library = 20 mg"),
                    # add Total for project
                    `Total units / biomass` = c("5 cells = 100 gr", "3 libraries = 18 gr", "1 library = 20 mg"),
                    #Add Number of plantlets
                    `Number plantlets` = c(ceiling(100/0.8), ceiling(18/0.8), ceiling(0.2/0.8))
                  )
colnames(dat) <- c("Type","Purpose","Unit/Biomass","Total units / biomass","Number of plantlets")
#Plot table
knitr::kable(dat, caption = "Summary of sequencing technologies, their purpose and requirements to complete the sagebrush genome project.")
```




# *In vitro* propagation of G2_b24_1

G2_b24_1 is among the top performers identified by @Barron2020. Here, we are propagating this individual line to produce the ca. 120 gr of biomass necessary for the sagebrush genome project (see Table \@ref(tab:tabproject)).

To predict how long it will take to produce the `r 120/0.8` plantlets for the project, we have developed the *propagationPred* R function. In addition to a time frame, this function provides estimates of labor time required to produce the growth and rooting media as well as the associated amount of media and required vessels (plates and Magenta vessels).    

  
## Applying *propagationPred* to G2_b24_1 {.tabset .tabset-pills}

### Analyses

The function has been applied with the following parameters (see section \@ref(predcode) for R code):

- `n`: 6 (plantlets).
- `ns`: 9.5 (on average after 10 weeks of growth).
- `r`: 0.93 (equal through generations). 
- `s`: 0.8 (equal through generations).
- `g`: 4 (generations).
- `biobank`: c(0,0,0.9,0). Meaning that 90% of plantlets will be biobanked at generation 3. 
- `date_user`: `r mdy("11/13/2020")-weeks(10)`.
- `growth_time_g`: c(weeks(10),weeks(8),weeks(8),weeks(8)).
- `biobanking_time`: weeks(15) (time required to get at 0.8 gr per plantlet).
- `rooting_time_g`: c(weeks(4),weeks(3),weeks(3),weeks(3)).
- `biomass_plantlet`: 0.8 (in gr).

The function outputs the table below allowing to prepare for key dates and tasks to successfully produce the biomass. 

```{r predictionprop, echo=FALSE, eval=TRUE}
#Source user-made function
source("Functions/Propagation.R")

#Apply propagationPred to top performer (G2_b24_1 from UTT2)
OUTdat <- propagationPred(n=6, ns=9.5, r=rep(0.93,4), s=rep(0.8,4), g=4, biobank=c(0,0,0.9,0), date_user=mdy("11/13/2020")-weeks(10), growth_time_g=c(weeks(10),weeks(8),weeks(8),weeks(8)), biobanking_time = weeks(15), rooting_time_g=c(weeks(4),weeks(3),weeks(3),weeks(3)), biomass_plantlet = 0.8)

#Subset OUTdat to only contain rows of interests
OUTdat <- OUTdat[1:11,]

#Plot table
DT::datatable(OUTdat, extensions = 'Buttons', options = list(dom = 'Blfrtip', buttons = c('copy', 'csv', 'excel', 'pdf', 'print')))
```

To better visualize the project, a plot (see Figure \@ref(fig:plotpred)) was produced and here are the key dates:

```{r plotpred, echo=FALSE, eval=TRUE, out.width="100%", fig.align='center', fig.cap= "Plot showing the time frame to conduct the in vitro propagation of G2_b24_1 to produce the biomass for the Sagebrush Genome Project and maintain the individual line."}
###~~~
#Plot data
###~~~
#pdf("Schedule_sagebrush_propagation_G2_b24_1.pdf")
#Populate plot
Type <- levels(OUTdat$Type)[c(3,2,4,1)]
colSeg <- c("black","green","blue","orange")

#Create empty plot
plot(x=c(as.Date(OUTdat$Date_start)), y= c(as.numeric(as.vector(OUTdat$N_plant_start))), 
     xlim= c(min(as.Date(OUTdat$Date_start)), max(as.Date(OUTdat$Date_end))+20), 
     ylim=c(0, max(as.numeric(as.vector(OUTdat$N_plant_start)))+50), type='n', xlab="Time (days)", ylab = "Number of Plantlets") 

biomassTot <- 120
biomass_plantlet <- 0.8
#Add objective of N plantlets
abline(h=biomassTot/biomass_plantlet, col='grey')
text(x=as.Date(OUTdat$Date_start[1]), y=biomassTot/biomass_plantlet+10, labels=paste("Number of Plantlets for Project (N: ", biomassTot/biomass_plantlet, " = ", biomassTot, " gr. )", sep=""), adj=0, col="grey", cex=.65)

#Add segments
for(i in 1:length(Type)){
  tmp <- OUTdat[which(OUTdat$Type == Type[i]),]
  segments(x0=as.Date(tmp$Date_start), x1=as.Date(tmp$Date_end), y0=as.numeric(as.vector(tmp$N_plant_start)), y1=as.numeric(as.vector(tmp$N_plant_end)), col=colSeg[i], lwd=1.5)
  #Add points to better see beginning and end of phases
  points(x=as.Date(tmp$Date_start), y=as.numeric(as.vector(tmp$N_plant_start)), col=colSeg[i], pch=16, cex=.8)
  points(x=as.Date(tmp$Date_end), y=as.numeric(as.vector(tmp$N_plant_end)), col=colSeg[i], pch=16, cex=.8)
  
  if(Type[i] == "Cutting"){
    text(x=as.Date(tmp$Date_start), y=as.numeric(as.vector(tmp$N_plant_end))+20, labels = as.Date(tmp$Date_start), srt=45, adj=0.5, cex=.6)
  }
  if(Type[i] == "Rooting"){
    text(x=as.Date(tmp$Date_end), y=as.numeric(as.vector(tmp$N_plant_end))+8, labels = as.Date(tmp$Date_end), srt=45, adj=0, cex=.6)
  }
  if(Type[i] == "Growth"){
    #Start of experiment
    startExp <- which(tmp$Date_start == as.character(min(as.Date(tmp$Date_start))))
    text(x=as.Date(tmp$Date_start)[startExp], y=as.numeric(as.vector(tmp$N_plant_start))[startExp]+8, labels = as.Date(tmp$Date_start)[startExp], srt=90, adj=0, cex=.6)
    
    #End of experiment
    endExp <- which(tmp$Date_end == as.character(max(as.Date(tmp$Date_end))))
    text(x=as.Date(tmp$Date_end)[endExp], y=as.numeric(as.vector(tmp$N_plant_end))[endExp]+8, labels = as.Date(tmp$Date_end)[endExp], srt=90, adj=0, cex=.6)
  }
  if(Type[i] == "Biobanking"){
    #Add segment(s) showing when rooted shoot tips are split between growth and biobanking
    ystart <- as.numeric(as.vector(OUTdat$N_plant_start[which(OUTdat$Date_start == tmp$Date_start & OUTdat$Type == "Growth")]))
    yend <- as.numeric(as.vector(OUTdat$N_plant_end[which(OUTdat$Date_end == as.character(tmp$Date_start) & OUTdat$Type == "Rooting")]))
    segments(x0=as.Date(tmp$Date_start), x1=as.Date(tmp$Date_start), y0=ystart, y1=yend, lty=2) 
    text(x=as.Date(tmp$Date_start)-4, y=ystart, paste("Split rooted shoot tips:", paste(as.character(tmp$Type), " (", tmp$N_plant_start, ")", " & Growth (", ystart, ")", sep="")), adj=0, srt=90, cex=.6)
    
    #End of experiment
    endExp <- which(tmp$Date_end == as.character(max(as.Date(tmp$Date_end))))
    text(x=as.Date(tmp$Date_end)[endExp], y=as.numeric(as.vector(tmp$N_plant_end))[endExp]+8, labels = paste(as.Date(tmp$Date_end)[endExp], " (", as.character(tmp$`Total_Biomass (gr)`), " gr.)", sep=''), srt=90, adj=0, cex=.6)
  }
  
}

#Add legend
legend("topleft", legend = Type, lty=1, col=colSeg, cex=.6)
#dev.off()
```

### R code {#predcode}

```{r predictionpropcode, echo=TRUE, eval=FALSE}
#Source user-made function
source("Functions/Propagation.R")

#Apply propagationPred to top performer (G2_b24_1 from UTT2)
OUTdat <- propagationPred(n=6, ns=9.5, r=rep(0.93,4), s=rep(0.8,4), g=4, biobank=c(0,0,0.9,0), date_user=mdy("11/13/2020")-weeks(10), growth_time_g=c(weeks(10),weeks(8),weeks(8),weeks(8)), biobanking_time = weeks(15), rooting_time_g=c(weeks(4),weeks(3),weeks(3),weeks(3)), biomass_plantlet = 0.8)

#Subset OUTdat to only contain rows of interests
OUTdat <- OUTdat[1:11,]

#Plot table
DT::datatable(OUTdat, extensions = 'Buttons', options = list(dom = 'Blfrtip', buttons = c('copy', 'csv', 'excel', 'pdf', 'print')))

###~~~
#Plot data
###~~~
#pdf("Schedule_sagebrush_propagation_G2_b24_1.pdf")
#Populate plot
Type <- levels(OUTdat$Type)[c(3,2,4,1)]
colSeg <- c("black","green","blue","orange")

#Create empty plot
plot(x=c(as.Date(OUTdat$Date_start)), y= c(as.numeric(as.vector(OUTdat$N_plant_start))), 
     xlim= c(min(as.Date(OUTdat$Date_start)), max(as.Date(OUTdat$Date_end))+20), 
     ylim=c(0, max(as.numeric(as.vector(OUTdat$N_plant_start)))+50), type='n', xlab="Time (days)", ylab = "Number of Plantlets") 

biomassTot <- 120
biomass_plantlet <- 0.8
#Add objective of N plantlets
abline(h=biomassTot/biomass_plantlet, col='grey')
text(x=as.Date(OUTdat$Date_start[1]), y=biomassTot/biomass_plantlet+10, labels=paste("Number of Plantlets for Project (N: ", biomassTot/biomass_plantlet, " = ", biomassTot, " gr. )", sep=""), adj=0, col="grey", cex=.65)

#Add segments
for(i in 1:length(Type)){
  tmp <- OUTdat[which(OUTdat$Type == Type[i]),]
  segments(x0=as.Date(tmp$Date_start), x1=as.Date(tmp$Date_end), y0=as.numeric(as.vector(tmp$N_plant_start)), y1=as.numeric(as.vector(tmp$N_plant_end)), col=colSeg[i], lwd=1.5)
  #Add points to better see beginning and end of phases
  points(x=as.Date(tmp$Date_start), y=as.numeric(as.vector(tmp$N_plant_start)), col=colSeg[i], pch=16, cex=.8)
  points(x=as.Date(tmp$Date_end), y=as.numeric(as.vector(tmp$N_plant_end)), col=colSeg[i], pch=16, cex=.8)
  
  if(Type[i] == "Cutting"){
    text(x=as.Date(tmp$Date_start), y=as.numeric(as.vector(tmp$N_plant_end))+20, labels = as.Date(tmp$Date_start), srt=45, adj=0.5, cex=.6)
  }
  if(Type[i] == "Rooting"){
    text(x=as.Date(tmp$Date_end), y=as.numeric(as.vector(tmp$N_plant_end))+8, labels = as.Date(tmp$Date_end), srt=45, adj=0, cex=.6)
  }
  if(Type[i] == "Growth"){
    #Start of experiment
    startExp <- which(tmp$Date_start == as.character(min(as.Date(tmp$Date_start))))
    text(x=as.Date(tmp$Date_start)[startExp], y=as.numeric(as.vector(tmp$N_plant_start))[startExp]+8, labels = as.Date(tmp$Date_start)[startExp], srt=90, adj=0, cex=.6)
    
    #End of experiment
    endExp <- which(tmp$Date_end == as.character(max(as.Date(tmp$Date_end))))
    text(x=as.Date(tmp$Date_end)[endExp], y=as.numeric(as.vector(tmp$N_plant_end))[endExp]+8, labels = as.Date(tmp$Date_end)[endExp], srt=90, adj=0, cex=.6)
  }
  if(Type[i] == "Biobanking"){
    #Add segment(s) showing when rooted shoot tips are split between growth and biobanking
    ystart <- as.numeric(as.vector(OUTdat$N_plant_start[which(OUTdat$Date_start == tmp$Date_start & OUTdat$Type == "Growth")]))
    yend <- as.numeric(as.vector(OUTdat$N_plant_end[which(OUTdat$Date_end == as.character(tmp$Date_start) & OUTdat$Type == "Rooting")]))
    segments(x0=as.Date(tmp$Date_start), x1=as.Date(tmp$Date_start), y0=ystart, y1=yend, lty=2) 
    text(x=as.Date(tmp$Date_start)-4, y=ystart, paste("Split rooted shoot tips:", paste(as.character(tmp$Type), " (", tmp$N_plant_start, ")", " & Growth (", ystart, ")", sep="")), adj=0, srt=90, cex=.6)
    
    #End of experiment
    endExp <- which(tmp$Date_end == as.character(max(as.Date(tmp$Date_end))))
    text(x=as.Date(tmp$Date_end)[endExp], y=as.numeric(as.vector(tmp$N_plant_end))[endExp]+8, labels = paste(as.Date(tmp$Date_end)[endExp], " (", as.character(tmp$`Total_Biomass (gr)`), " gr.)", sep=''), srt=90, adj=0, cex=.6)
  }
  
}

#Add legend
legend("topleft", legend = Type, lty=1, col=colSeg, cex=.6)
#dev.off()
```

# References

<div id="refs"></div>

# Appendix 1 {.appendix}

Citations of all R packages used to generate this report. 

```{r generateBibliography, eval = T, results="asis", cache = F, echo=F, warning = FALSE, message=FALSE}
library("knitcitations")
cleanbib()
options("citation_format" = "pandoc")
read.bibtex(file = "packages.bib")
``` 

# Appendix 2 {.appendix}

Version information about R, the operating system (OS) and attached or R loaded packages. This appendix was generated using `sessionInfo()`.

```{r eval=T, echo=F, warning = FALSE, message=FALSE}
sessionInfo()
```
