---
title: "Sagebrush Genome Project"
output:
  bookdown::html_document2: 
    toc: TRUE
    toc_float: TRUE
link-citations: yes
fontsize: 12pt
bibliography: References.bib
csl: AmJBot.csl
---


```{js logo-js, echo=FALSE}
$(document).ready(function() {
  $('#header').parent().prepend('<div id=\"logo\"><img src=\"Images/boisestate-primarylogo-2color-rgb.png\" style=\"position:absolute; top:25px; right:0; padding:10px; height:120px\"></div>');
  $('#header').css('margin-right', '120px')
});
```

# Aim

The Sagebrush Genome Project supports the overarching aim of the GEM3 project, which is to **understand genome to phenome (G2P) mechanisms that translate to adaptive capacity of sagebrush (*Artemisia tridentata*) populations.** More specifically, in this project, we will be focusing on assessing sagebrush adaptive capacity to respond to drought and heat stress caused by climate change. 

# Objectives

This project is subdivided into 5 objectives as follows:

1. **Individual lines:** Developing and applying an *in vitro* tissue culture approach to propagate sagebrush individual lines supporting objectives 2-4. This objective is key since preliminary analyses demonstrated that the sagebrush genome is highly out-bred. For these reasons, we are aiming at not only produce enough biomass of the same individual line to produce the reference genome, but also maintain individual lines to support genotype-by-environment (GxE) experiments (objective 4).
2. **Reference genome:** Sequencing, phasing and annotating the diploid genome of basin big sagebrush (*Artemisia tridentata* subsp. *tridentata*). This taxon as 2n=2x=18 chromosomes and a 2C genome size of ca. 9.5Gbp [see @Barron2020]. To deliver such complex genome, we aim at generating 50-100X raw data coverage (haploid sagebrush genome size: 4.5 Gbp) using PacBio technology and phase it using proximity ligation data. Such wet-lab procedures are requiring the production of ca. **120 gr. of leaf biomass**. Annotation of the reference genome will be conducted based on RNA-seq data generated during objective 4 (through GxE experiments). This will allow focusing on annotating genes involved in responding to drought and heat stresses and therefore assess G2P mechanisms. 
3. **Comparative genomics:** To further investigate genomic differences between sagebrush populations, a re-sequencing approach coupled with pan-genomic analyses will be applied to compare genomes from individual lines growing in contrasting climates (in terms of drought and heat stresses).
4. **Common gardens:** GxE experiments focusing on measuring the phenotypic and transcriptomic responses of sagebrush individual lines to drought, heat and drought+heat stresses will be conducted. Using clones will allow to disentangle respective contributions of genomic vs. phenotypic plasticty processes on the adaptive capacity of sagebrush genotypes. An overarching objective here is to identify adaptive loci key in drought, heat and drought+heat responses. These adaptive loci will then be used to model sagebrush adaptive capacity at population level in objective 5.
5. **Landscape genomics:** Apply adaptive loci (sequenced using a baiting approach) on sagebrush individuals collected from populations representing climatic and geographic range of the species. First, field collections will be biobanked (= conducting DNA extractions and establishing a seed bank) and herbarium specimens digitized. The age of each collected individual will be estimated and compared to onset of megadrought in the US [@williams2020Science]. This preliminary data will provide hypotheses on the establishment of individuals and populations in the landscape and the potential effects of drought on the gene pool. We hypothesize that the effect of megadrought on established individuals will be less dramatic than on those establishing post megadrought. This means that if sagebrush megadrought (and associated heat) response is genetically driven then more genetic diversity (based on adaptive loci inferred in objective 4) will be observed for individuals established prior to megadrought, whereas those established post megadrought will show a strong genetic bottleneck effect corresponding to natural selection processes acting upon sageberush populations.     

# Timetable

The timetable associated to our objectives is displayed in Figure \@ref(fig:timetab).

```{r timetab, warning = F, echo=F, message = F, results="asis", fig.cap= "Timetable associated to production of data for the sagebrush genome project. Please see text for more details.", out.width = '100%', cache = F}
#Timetable

timetab <- data.frame("Objective" = c(rep("Reference genome", 2), rep("Comparative genomics",2), rep("Common gardens", 4), rep("Landscape genomics",4), rep("Individual lines", 2)), "Task" = c("Seq., phasing & assembly", "Annotation", "Re-sequencing","Pan-genomics","GxE exp.", "RNA-seq", "Transcriptomics", "Id. adaptive loci", "Baiting", "Modeling", "Fieldwork", "Biobanking & Digitization","Propagation", "Maintaining lines"), "Start_Date" = lubridate::mdy(rep("NA", 14)), "End_Date" = lubridate::mdy(rep("NA", 14)))

###
#Ref. genome
# G1_b2
###

#Harvesting biomass?
timetab$Start_Date[1] <- as.Date("04-07-21", tryFormats = c("%m-%d-%y")) 
#Sequencing done
timetab$End_Date[1] <- timetab$Start_Date[1] + lubridate::weeks(28)

#Genome annotation start
timetab$Start_Date[2] <- timetab$End_Date[1]
#Genome annotation end
timetab$End_Date[2] <- timetab$Start_Date[2] + lubridate::weeks(16)

#Comparative genomics (incl. re-sequencing)
#Re-sequencing
timetab$Start_Date[3] <- as.Date("04-30-21", tryFormats = c("%m-%d-%y"))
timetab$End_Date[3] <- timetab$Start_Date[3] + lubridate::weeks(16)
#Pan-genomics
timetab$Start_Date[4] <- timetab$End_Date[3]
timetab$End_Date[4] <- timetab$End_Date[2]

###
#Common gardens
# G1_b2, G2_b27
###

#When ready for GxE
timetab$Start_Date[5] <- as.Date("04-01-21", tryFormats = c("%m-%d-%y")) + lubridate::weeks(2)

#When GxE end?
timetab$End_Date[5] <- as.Date("04-09-21", tryFormats = c("%m-%d-%y")) + lubridate::weeks(28)

#RNA-Seq
timetab$Start_Date[6] <- as.Date("04-01-21", tryFormats = c("%m-%d-%y")) + lubridate::weeks(20)
timetab$End_Date[6] <- timetab$End_Date[5] + lubridate::weeks(10)

#Transcriptomics
timetab$Start_Date[7] <- timetab$Start_Date[6] + lubridate::weeks(10)
timetab$End_Date[7] <- timetab$End_Date[6] + lubridate::weeks(20)

#Identify adaptive loci
timetab$Start_Date[8] <- timetab$Start_Date[7]
timetab$End_Date[8] <- timetab$End_Date[6] + lubridate::weeks(8)

###
#Landscape genomics
# Population sampling
###

#Baiting
timetab$Start_Date[9] <- timetab$End_Date[8]
timetab$End_Date[9] <- timetab$Start_Date[9] + lubridate::weeks(10)

#Modeling
timetab$Start_Date[10] <- timetab$End_Date[9]
timetab$End_Date[10] <- timetab$Start_Date[10] + lubridate::weeks(20)

#Fieldwork
timetab$Start_Date[11] <- as.Date("10-26-20", tryFormats = c("%m-%d-%y"))
timetab$End_Date[11] <- as.Date("11-19-20", tryFormats = c("%m-%d-%y"))

#DNA bank
timetab$Start_Date[12] <- as.Date("06-01-21", tryFormats = c("%m-%d-%y"))
timetab$End_Date[12] <- timetab$Start_Date[9]
  
###
#Individual lines
###
#Propagation
timetab$Start_Date[13] <- as.Date("02-13-20", tryFormats = c("%m-%d-%y"))
timetab$End_Date[13] <- as.Date("04-07-21", tryFormats = c("%m-%d-%y"))
#Maintaining lines
timetab$Start_Date[14] <- timetab$End_Date[13] + lubridate::weeks(1)
timetab$End_Date[14] <- timetab$End_Date[10]

#Add col with y height
timetab$Y <- c(rep(2.625, 2), rep(2.250, 2), 1.875, 1.500, 1.125, 0.750, rep(0.375, 4), rep(3,2))
#Add color
# Blue = wet-lab
# orange = bioinformatics
# brown = wet/bioinfo
timetab$col <- c("brown", "orange", "blue", "orange", "blue", "blue", "orange", "orange", "blue", "orange", "blue", "brown","blue", "blue")

#Add duration (in months)
timetab$duration <- round(lubridate::time_length(x = lubridate::interval(timetab$Start_Date, timetab$End_Date), "month"),1)

###
#PLOT
###

#pdf(file = "Figures/timetable_whole_project.pdf")
#Plot timetable
plot(x=c(lubridate::floor_date(min(timetab$Start_Date), "month")-10, lubridate::round_date(max(timetab$End_Date), "month")), y= c(0,3.5), type="n", axes = F, ylab = "", xlab = "")

#Abline to separate years
abline(v =  as.Date("12-31-20", tryFormats = c("%m-%d-%y")))
abline(v =  as.Date("12-31-21", tryFormats = c("%m-%d-%y")))
#2020
text(x = as.Date(timetab$Start_Date[11]) + abs(floor((as.Date(timetab$Start_Date[11])-as.Date("12-31-20", tryFormats = c("%m-%d-%y")))/2)), y = 0.01, cex = 0.8, "2020")
#2021
text(x = as.Date("01-01-21", tryFormats = c("%m-%d-%y")) + abs(floor((as.Date("01-01-21", tryFormats = c("%m-%d-%y"))-as.Date("12-31-21", tryFormats = c("%m-%d-%y")))/2)), y = 0.01, cex = 0.8, "2021")
#2022
text(x = as.Date("01-01-22", tryFormats = c("%m-%d-%y")) + abs(floor(as.Date("01-01-22", tryFormats = c("%m-%d-%y")) - as.Date(max(timetab$End_Date)))/2), y = 0.01, cex = 0.8, "2022")

#Add customed axes
#Y axis = Individual lines
axis(side = 2, at = seq(from = 0, to = 3, by =3/8)[c(2, 6:9)], las = 2, labels = rev(gsub(" ", "\n", unique(timetab$Objective))[c(5,1,2,3,4)]), cex.axis= .7, pos=lubridate::floor_date(min(timetab$Start_Date), "month"), adj=0)

#X axis = time
datespan <- seq(from = lubridate::floor_date(min(timetab$Start_Date), "month"), to = lubridate::round_date(max(timetab$End_Date), "month"), "month")

axis(side = 1, at = datespan, labels = as.vector(lubridate::month(datespan, label = TRUE, abbr = T)))

#Add titles to axes
title(main="", xlab="Time", ylab="")

#Add segments showing tasks
segments(x0 = as.Date(timetab$Start_Date), x1 = as.Date(timetab$End_Date), y0 = as.numeric(timetab$Y), y1 = as.numeric(timetab$Y), lwd = 2, col = timetab$col)

#Print tasks next to arrows
text(x = as.Date(timetab$Start_Date) + floor((as.Date(timetab$End_Date)-as.Date(timetab$Start_Date))/2)  , y = timetab$Y+0.15, paste(as.vector(timetab$Task), "\n", "(", timetab$duration, " m.)", sep=""), srt=8, cex=.7)

#Arrows with fiscal years
#Y2
arrows(x0 = lubridate::floor_date(min(timetab$Start_Date), "month"), x1 = as.Date("09-30-20", tryFormats = c("%m-%d-%y")), y0 = 0.18, y1 = 0.18, length = 0.1, angle = 10)
text(x = lubridate::floor_date(min(timetab$Start_Date), "month") + floor((as.Date("09-30-20", tryFormats = c("%m-%d-%y"))-lubridate::floor_date(min(timetab$Start_Date), "month"))/2), y = 0.27, "Year 2", cex = 0.8)

#Y3
arrows(x0 = as.Date("10-01-20", tryFormats = c("%m-%d-%y")), x1 = as.Date("09-30-21", tryFormats = c("%m-%d-%y")), y0 = 0.18, y1 = 0.18, length = 0.1, angle = 10, code = 3)
text(x = as.Date("10-01-20", tryFormats = c("%m-%d-%y")) + floor((as.Date("09-30-21", tryFormats = c("%m-%d-%y"))-as.Date("10-01-20", tryFormats = c("%m-%d-%y")))/2), y = 0.27, "Year 3", cex = 0.8)

#Y4
arrows(x0 = as.Date("10-01-21", tryFormats = c("%m-%d-%y")), x1 = lubridate::ceiling_date(max(timetab$End_Date), "month"), y0 = 0.18, y1 = 0.18, length = 0.1, angle = 10, code = 1)
text(x = as.Date("10-01-21", tryFormats = c("%m-%d-%y")) + abs(floor((as.Date("10-01-21", tryFormats = c("%m-%d-%y"))-lubridate::ceiling_date(max(timetab$End_Date), "month"))/2)), y = 0.27, "Year 4", cex = 0.8)

#Add connections between tasks
# transcriptomics to annotation
arrows(x0 = timetab$Start_Date[7]+55, x1 = timetab$Start_Date[7]+55, y0 = timetab$Y[7], y1 = timetab$Y[2], score = 1, length = 0.1, lwd = 1.5, lty = 6, angle = 20, col = "black", code = 3)

#Annotation to pan-genomics
arrows(x0 = timetab$Start_Date[7]+55, x1 = timetab$Start_Date[7]+55, y0 = timetab$Y[2], y1 = timetab$Y[4], score = 1, length = 0.1, lwd = 1.5, lty = 6, angle = 20, col = "black", code = 3)

#pan-genomics to transcriptomics
arrows(x0 = timetab$Start_Date[7]+55, x1 = timetab$Start_Date[7]+55, y0 = timetab$Y[7], y1 = timetab$Y[4], score = 1, length = 0.1, lwd = 1.5, lty = 6, angle = 20, col = "black", code = 3)


#Pan-genomics to adaptive loci
arrows(x0 = timetab$End_Date[2]-20, x1 = timetab$End_Date[2]-20, y0 = timetab$Y[4], y1 = timetab$Y[8], score = 1, length = 0.1, lwd = 1.5, lty = 6, angle = 20, col = "black")

#Transcriptomics to adaptive loci
arrows(x0 = timetab$Start_Date[7]+5, x1 = timetab$Start_Date[7]+5, y0 = timetab$Y[7], y1 = timetab$Y[8], score = 1, length = 0.1, lwd = 1.5, lty = 6, angle = 20, col = "black")


#Propagation to sequencing
arrows(x0 = timetab$End_Date[13]-40, x1 = timetab$Start_Date[1]-2, y0 = timetab$Y[13], y1 = timetab$Y[1], score = 1, length = 0.1, lwd = 1.5, lty = 6, angle = 20, col = "black")

#Propagation to Re-sequencing
arrows(x0 = timetab$End_Date[13]-50, x1 = timetab$Start_Date[3]-2, y0 = timetab$Y[13], y1 = timetab$Y[3], score = 1, length = 0.1, lwd = 1.5, lty = 6, angle = 20, col = "black")

#Propagation to GxE
arrows(x0 = timetab$End_Date[13]-60, x1 = timetab$Start_Date[5]-2, y0 = timetab$Y[13], y1 = timetab$Y[5], score = 1, length = 0.1, lwd = 1.5, lty = 6, angle = 20, col = "black")

#Adaptive loci to Baiting
arrows(x0 = timetab$End_Date[8], x1 = timetab$Start_Date[9], y0 = timetab$Y[8], y1 = timetab$Y[9], score = 1, length = 0.1, lwd = 1.5, lty = 6, angle = 20, col = "black")

# Add a legend
legend("bottomleft", 
  legend = c("Wet-lab", "Bioinformatics", "Wet-lab/Bioinfo.", "Data connectivity"), 
  col = c("blue","orange", "brown", "black"), 
  lwd = rep(2, 4),
  lty = c(rep(1, 3), 6),
  horiz = F , 
  inset = c(0.07, 0.2),
  cex = 0.6)
#invisible(dev.off())

#knitr::include_graphics("Figures/timetable_whole_project.jpg")
```

# Project structure

Each of the objective described above and in Figure \@ref(fig:timetab) are further detailed in the following webpage:

- **[Propagation program](Individual_lines.html)**: This webpage covers the tissue culture methods applied in this project together with data on individual lines maintained at BSU.
- **[Sequencing](Sequencing.html):** This webpage describes the approach used to sequence, phase and annotate sagebrush genome.
- **[Annotation](Annotation.html):** This webpage provides insights into the GxE experiments used to estimate the effect of drought, heat and drought+heat on sagebrush individual lines. RNA-seq data produced by these experiments will be used to annotate sagebrush genome and identify adaptive loci.

# References