<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Sagebrush Genome Project</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
      </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Sagebrush Genome Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Biomass_production.html">Biomass production</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Sagebrush Genome Project</h1>

</div>


<div id="aim" class="section level1">
<h1><span class="header-section-number">1</span> Aim</h1>
<p>Sequencing, phasing and annotating the diploid genome of basin big sagebrush (<em>Artemisia tridentata</em> subsp. <em>tridentata</em>). This taxon as 2n=2x=18 chromosomes and a 2C genome size of 9.5Gbp. In addition, preliminary analyses demonstrated that the genome is highly out-bred.</p>
</div>
<div id="overall-approach-requirements" class="section level1">
<h1><span class="header-section-number">2</span> Overall approach &amp; Requirements</h1>
<p>To be able to phase such complex genome, we aim at generating 50-100X raw data coverage (haploid sagebrush genome size: 4.5 Gbp) using PacBio technology and phase it using proximity ligation data. Such wet-lab procedures are requiring the production of ca. <strong>120 gr. of leaf biomass</strong>.</p>
<div id="biomass-requirements-for-genome-project" class="section level2">
<h2><span class="header-section-number">2.1</span> Biomass requirements for genome project</h2>
<p>We aim at applying newly developed <em>in vitro</em> tissue culture technique <span class="citation">(Barron et al. <a href="#ref-Barron2020">2020</a>)</span> to G2_b24_1 (from UTT2, drought-sensitive genotype; see Figure <a href="#fig:plantlets">2.1</a>) to produce the ca. 120 gr of leaf biomass necessary for genome sequencing, phasing and annotation.</p>
<div class="figure" style="text-align: center"><span id="fig:plantlets"></span>
<img src="Figures/G2_b20_1_15wk_old_plantlet.jpg" alt="Example of a 15-week old plantlet grown in vitro in Magenta vessel." width="50%" />
<p class="caption">
Figure 2.1: Example of a 15-week old plantlet grown in vitro in Magenta vessel.
</p>
</div>
<p>After 15 weeks of <em>in vitro</em> culture, <strong>a plantlet produces 0.8 gr. of leaf biomass</strong> that can be used for the genome project (Figure <a href="#fig:plantletweight">2.2</a>). To meet our requirements <strong>a minimum of 150 plantlets have to be produced at BSU</strong> (see table below). The biomass can be flash frozen and stored in the -80C freezer. The estimation of biomass does not account for a DNA extraction trial to optimize protocol (we need at least 15ug of HMW DNA for PacBio sequencing). We will have to discuss with the project manager at Dovetail Genomics.</p>
<div class="figure" style="text-align: center"><span id="fig:plantletweight"></span>
<img src="Figures/G2_b20_1_Biomass_0.855g_after2min.jpg" alt="A 15-week old plantlet grown in vitro in Magenta vessel generates 0.8 gr of leaf biomass for sequencing." width="50%" />
<p class="caption">
Figure 2.2: A 15-week old plantlet grown in vitro in Magenta vessel generates 0.8 gr of leaf biomass for sequencing.
</p>
</div>
<div id="number-of-plantlets-per-sequencing-technology" class="section level3">
<h3><span class="header-section-number">2.1.1</span> Number of plantlets per sequencing technology</h3>
<p>Table <a href="#tab:tabproject">2.1</a> provides a summary of sequencing technologies, their purpose and requierements to complete the sagebrush genome project.</p>
<table>
<caption><span id="tab:tabproject">Table 2.1: </span>Summary of sequencing technologies, their purpose and requierements to complete the sagebrush genome project.</caption>
<colgroup>
<col width="32%" />
<col width="17%" />
<col width="18%" />
<col width="16%" />
<col width="14%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Type</th>
<th align="left">Purpose</th>
<th align="left">Unit/Biomass</th>
<th align="left">Total units / biomass</th>
<th align="right">Number of plantlets</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">PacBio sequencing</td>
<td align="left">De novo genome assembly</td>
<td align="left">1 cell = 20 gr</td>
<td align="left">5 cells = 100 gr</td>
<td align="right">125</td>
</tr>
<tr class="even">
<td align="left">Proximity ligation (Hi-C for phasing genome)</td>
<td align="left">Phasing</td>
<td align="left">1 Illumina library = 6 gr</td>
<td align="left">3 libraries = 18 gr</td>
<td align="right">23</td>
</tr>
<tr class="odd">
<td align="left">RNA-seq</td>
<td align="left">Annotation</td>
<td align="left">1 library = 20 mg</td>
<td align="left">1 library = 20 mg</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="meet-the-team" class="section level1">
<h1><span class="header-section-number">3</span> Meet the team</h1>
<ul>
<li><strong>Peggy Martinez, M.S.</strong> (<a href="mailto:peggymartinez@boisestate.edu" class="email">peggymartinez@boisestate.edu</a>): Laboratory Technician leading the sagebrush <em>in vitro</em> culture program at Boise State University.</li>
<li><strong>Rachael Barron, M.S.</strong> (<a href="mailto:rachael.barron@simplot.com" class="email">rachael.barron@simplot.com</a>): Associate Scientist at Simplsot who has contributed her expertise on <em>in vitro</em> culture to this project.</li>
<li><strong>Marcelo Serpe, Ph.D</strong>: Professor at Boise State University.</li>
<li><strong>Sven Buerki, Ph.D</strong> (<a href="mailto:sven.buerki@boisestate.edu" class="email">sven.buerki@boisestate.edu</a>): Assistant Professor at Boise State University and PI of this project funded by <a href="funding.html">GEM3</a>.</li>
</ul>
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references">
<div id="ref-Barron2020">
<p>Barron, Rachael, Peggy Martinez, Marcelo Serpe, and Sven Buerki. 2020. “Development of an in Vitro Method of Propagation for Artemisia Tridentata Subsp. Tridentata to Support Genome Sequencing and Genotype-by-Environment Research.” <em>Plants</em> 9 (12). <a href="https://doi.org/10.3390/plants9121717">https://doi.org/10.3390/plants9121717</a>.</p>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
