<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Protocols</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Sagebrush Genome Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Individual_lines.html">Propagation</a>
</li>
<li>
  <a href="Biomass_production.html">Biomass</a>
</li>
<li>
  <a href="Sequencing.html">Sequencing</a>
</li>
<li>
  <a href="Annotation.html">Annotation</a>
</li>
<li>
  <a href="Protocols.html">Protocols</a>
</li>
<li>
  <a href="Team.html">Team</a>
</li>
<li>
  <a href="funding.html">Acknowledgements</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Protocols</h1>

</div>


<div id="chromosome-counting" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Chromosome counting</h1>
<div id="objective" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Objective</h2>
<p>Counting chromosomes of <em>Artemisia tridentata</em> using root meristems of individual lines maintained <em>in vitro</em> at BSU. This approach will help ascertaining the ploidy level (2n) of our individual lines.</p>
</div>
<div id="material-reagents" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Material &amp; reagents</h2>
<p><strong>Day 1</strong></p>
<ul>
<li>Little cooler for getting ice (room XXX).</li>
<li>Labeled glass large Petri dishes.</li>
<li>Tweezers (to pull roots out of Magenta vessels).</li>
<li>Distilled water.</li>
<li>Space in the fridge to store at 4<span>°</span>C for at least 10 hours (can do 24 hours).</li>
</ul>
</div>
<div id="plant-material" class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> Plant material</h2>
<p>Young whitish roots are sampled from individuals cultured <em>in vitro</em> either at <span style="color: black;font-weight:bold">Growth</span> or <span style="color: blue;font-weight:bold">Rooting</span> phases.</p>
</div>
<div id="sampling-and-fixing-tissue" class="section level2" number="1.4">
<h2><span class="header-section-number">1.4</span> Sampling and fixing tissue</h2>
<ol style="list-style-type: decimal">
<li>Sample young whitish roots and place them in a Petri dish.</li>
<li>Use achromatic spindle inhibitors to stop mitotic divisions (at metaphase) and contract chromosomes. This can be done following two approaches:
<ul>
<li>Treat roots with saturated solution of water and <span>α</span>-monobromonaphtalene for 3 hours at room temperature. Start by homogenizing the stock solution and adding supernatant to cover roots in the Petri dish.</li>
<li>Treat roots in a mix of water and crushed ice (4<span>°</span>C) for 10 hours.</li>
</ul></li>
<li>Plant tissue fixation and conserving chromosomes.
<ul>
<li>Prepare a falcon tube with the solution of ethanol:acetic acid (3:1). Add few drops of the acetocarmin solution and one drop of iron acetate. Homogenize the solution by gently shaking the falcon tube.</li>
<li>Transfer roots into falcon tube and store at room temperature for 1-2 days.</li>
</ul></li>
</ol>
</div>
<div id="staining-tissue" class="section level2" number="1.5">
<h2><span class="header-section-number">1.5</span> Staining tissue</h2>
<ol start="4" style="list-style-type: decimal">
<li>Pour acetocarmin solution in a porcelain crucible and add roots. Cover crucible with a lid.</li>
<li>Prepare a Petri dish filled with 45% acetic acid. This will be used to cool down stained roots and preserve them until chromosome numbers is established.</li>
<li>Gently heat solution in porcelain crucible with a Bunsen burner for ca. 2 minutes. Avoid boiling! This will stain chromosomes in red.</li>
<li>Pour content of porcelain crucible into Petri dish.</li>
</ol>
</div>
<div id="squashing-tissue" class="section level2" number="1.6">
<h2><span class="header-section-number">1.6</span> Squashing tissue</h2>
<ol start="8" style="list-style-type: decimal">
<li>Put a drop of the solution in the Petri dish on a microscope slide and add a root tip in it.</li>
<li>Remove the root cap under the scope using tweezers.This will facilitate the squash and observation of chromosomes.</li>
<li>Cover with a plastic coverslip and use handle of tweezers to squash the tissue. Go gently!</li>
<li>Observe under microscope and take picture to validate chromosome count.</li>
</ol>
</div>
<div id="preparing-stock-solutions" class="section level2" number="1.7">
<h2><span class="header-section-number">1.7</span> Preparing stock solutions</h2>
<p>Recipes to prepare solutions presented here were adapted from this <a href="Data/Caryologie_methodes_INRA_091118.pdf">document</a> (in french).</p>
<div id="ethanolacetic-acid-solution" class="section level3" number="1.7.1">
<h3><span class="header-section-number">1.7.1</span> Ethanol:acetic acid solution</h3>
<ul>
<li>Mix one volume of acetic acid for three volumes of ethanol. Store at 4<span>°</span>C.</li>
</ul>
</div>
<div id="acetocarmin-solution" class="section level3" number="1.7.2">
<h3><span class="header-section-number">1.7.2</span> Acetocarmin solution</h3>
<p>For 1000 mL for acetocarmin solution:</p>
<ol style="list-style-type: decimal">
<li>Measure 450 mL of 45% glacial acetic acid and pour in <strong>XXX</strong> (a vessel that we can heat). See Figure in this <a href="Data/Carmin_acétique_acétate_de_fer.pdf">document</a>.</li>
<li>Measure 550 mL of distilled water and add to vessel.</li>
<li>Weigh 5 gr. of carmin (Fisher C579-25). Carmine is a basic dye that is prepared from the insect <em>Coccus cacti</em>.</li>
<li>Add boileezers, and reflux for 2-3 hours.</li>
<li>Filter into dark bottles and store at 4<span>°</span>C.</li>
</ol>
<p>This solution can be stored for a long time. Staining can be intensified by adding ferric chloride (FeCl<sub>2</sub>·6H<sub>2</sub>O); add 5 mL of a 10 % ferric chloride solution per 100 mL of % acetocarmine.</p>
<p>Kansas State University as also put together a slightly different protocol for the acetocarmin solution: <a href="https://www.k-state.edu/wgrc/electronic_lab/aceto_stain.html" class="uri">https://www.k-state.edu/wgrc/electronic_lab/aceto_stain.html</a></p>
</div>
</div>
</div>
<div id="flow-cytometry" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Flow cytometry</h1>
<div id="objective-1" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Objective</h2>
<p>Estimate 2C genome size of <em>Artemisia tridentata</em> based on leaf material using the one-step protocol with general purpose buffer (GPB) by <span class="citation"><a href="#ref-Loureiro2007" role="doc-biblioref">Loureiro et al.</a> (<a href="#ref-Loureiro2007" role="doc-biblioref">2007</a>)</span> and the protocol described in <span class="citation"><a href="#ref-Pellicer2014" role="doc-biblioref">Pellicer and Leitch</a> (<a href="#ref-Pellicer2014" role="doc-biblioref">2014</a>)</span>. The publications are accessible <a href="Data/">here</a>.</p>
<p>The flow cytometry will be conducted at BSU using a <a href="Data/bd_accuri_optical_filter_guide.pdf">BD Accuri™ C6 Flow Cytometer</a>.</p>
</div>
<div id="challenge" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Challenge</h2>
<p>One thing that we need to bear in mind is that the coefficient of variation of the fluorescence peaks should be below 5%, and that the measurements should be taken using a linear scale rather that a logarithmic scale.</p>
</div>
<div id="plant-material-1" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Plant material</h2>
<p>Genome size is estimated using <strong>young, fresh leaves for both our target (sagebrush) and internal standards</strong>.</p>
<p>Concerning reference internal standards, we recommend that several species, covering a broad range of genome sizes, are kept growing in the laboratory to enable the most appropriate standard to be selected for each particular analysis.</p>
<div id="internal-standards" class="section level3" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> Internal standards</h3>
<p>Check what seeds we have in the lab.:</p>
<ul>
<li><em>Petroselinum crispum</em> ‘Curled moss’ (2C=4.5 pg).</li>
<li><em>Pisum sativum</em> ‘Ctirad’ (2C=9.09 pg).</li>
<li><em>Solanum lycopersicum</em> L. ‘Stupiké polní rane’ (2C=2 pg).</li>
</ul>
</div>
</div>
<div id="protocol" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Protocol</h2>
<p>The recipe of the buffer and the steps are described in the paper, but briefly:</p>
<ul>
<li>Work under cold conditions (keep samples and buffers on ice).</li>
<li>Cut a small portion of leaf tissue (1cm<sup>2</sup>, or less) from both the target and the internal standard.</li>
<li>Place samples in a small petri dish with 1ml of ice-cold buffer.</li>
<li>Chop with a sharp new razor blade till obtaining a fine lysate.</li>
<li>Add another 1ml of buffer.</li>
<li>Filter through a 30 um nylon mesh.</li>
<li>Stain with 100 ul of 1mg/ml Propidium iodide. Nuclei were stained with 50 µg mL−1 propidium iodide (PI; Fluka, Buchs, Switzerland), and 50 µg mL−1 RNase (Sigma, St Louis, MO, USA) was added to nuclear suspension to prevent staining of double-stranded RNA.</li>
<li>Incubate 5-10 min on ice.</li>
<li>Measure on the flow cytometer (here the <a href="Data/bd_accuri_optical_filter_guide.pdf">BD Accuri™ C6 Flow Cytometer</a>).</li>
</ul>
</div>
<div id="preparing-buffer" class="section level2" number="2.5">
<h2><span class="header-section-number">2.5</span> Preparing buffer</h2>
<div id="general-composition-of-the-lysis-buffer" class="section level3" number="2.5.1">
<h3><span class="header-section-number">2.5.1</span> General composition of the lysis buffer</h3>
<p><strong>GBP:</strong> 0.5 mM spermine.4HCl, 30 mM sodium citrate, 20 mM MOPS (MOPS, 4-Morpholinepropane sulfonate), 80 mM KCl, 20 mM NaCl, 0.5 % (v/v) Triton X-100, pH 7.0.</p>
</div>
<div id="procedure-to-prepare-the-lysis-buffer-gbp" class="section level3" number="2.5.2">
<h3><span class="header-section-number">2.5.2</span> Procedure to prepare the lysis buffer (GBP)</h3>
<ol style="list-style-type: decimal">
<li>Weigh the following chemicals into a glass (final volume of 200 mL):
<ul>
<li>Spermine 0.5 mM (0.0348 g)</li>
<li>Sodium citrate 3mM (1.764 g)</li>
<li>MOPS 20 mM (0.836 g)</li>
<li>KCl 80 mM (1.193 g)</li>
<li>NaCl 20 mM (0.233 g)</li>
</ul></li>
<li>Add double distilled water to 200 mL and mix with a rotating magnet and dissolve the chemicals</li>
<li>Add 1 mL of Triton X-100 and mix until diluted</li>
<li>Adjust to pH 7</li>
<li>Add 6 g of PVP-40 and mix until dissolved (Add PVP while the buffer is being mixed as it tends to clog. Be patient and leave it with a rotating magnet until completely dissolved)</li>
<li>Store in the fridge or prepare aliquots of 50 mL which can be stored on the freezer. Use one each time and once thawed do not freeze again. It can be stored in the fridge for weeks.</li>
</ol>
<p>The buffer is now ready to use. If you notice that the sample becomes dark when prepared, add 200 ul of B-mercaptoethanol to the buffer and mix (or equivalent if aliquots have been prepared).</p>
</div>
</div>
</div>
<div id="managing-linux-accounts-and-sharing-folders" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Managing Linux accounts and sharing folders</h1>
<p>To more efficiently benefit from our Linux computers, we will provide overview of protocols to:</p>
<ul>
<li>Create users.</li>
<li>Delete users.</li>
<li>Create groups.</li>
<li>Add users to groups.</li>
<li>Create shared folders.</li>
<li>Assign shared folders to a group.</li>
<li>Switch user to check changes.</li>
</ul>
<p>In RStudio, you can send commands directly into the Terminal (instead of the Console) by using this combination of keys: <code>Ctrl + Alt + Enter</code>. The Terminal has to be selected before using this approach.</p>
<div id="create-users" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Create users</h2>
<p>For this purpose, we will be using the <code>adduser</code> command:</p>
<pre class="bash"><code>#Create user, add password and create home directory (without admin rights)
sudo adduser username

#For example
sudo adduser anthonymelton</code></pre>
<p>You will be asked to answer multiple questions, only the password is mandatory.</p>
<p>Complete creation of sub-directories for new user by typing these commands:</p>
<pre class="bash"><code>#Switch to the new user account (see below for more details)
su username

#Create sub-directories (e.g. Documents/, Downloads/, etc.)
xdg-user-dirs-update --force</code></pre>
<p>To view the list of users on the Linux computer type this command:</p>
<pre class="bash"><code>cut -d : -f 1 /etc/passwd</code></pre>
<p>To view the list of logged in users do:</p>
<pre class="bash"><code>w</code></pre>
<p>To <strong>provide admin (<code>sudo</code>) rights to a user</strong> that you created do the following:</p>
<pre class="bash"><code>#Create a new used with admin/sudo rights
sudo usermod -aG sudo username

#For example
sudo usermod -aG sudo anthonymelton</code></pre>
<p>On the other hand, <strong>if you want to reverse a user from <code>sudo</code> to <code>standard</code> rights</strong>, do the following:</p>
<pre class="bash"><code>#General syntax
sudo deluser username sudo</code></pre>
<p>This command will only remove user ‘username’ from the sudo group, but it will not delete the user permanently from the system. Now, they become regular user and cannot do any administrative tasks as sudo user.</p>
</div>
<div id="delete-users" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Delete users</h2>
<p>After someone leaves the lab, you might have to delete their user account. You can do that by using the following command, but please make sure that you have archived all their data and code prior to proceeding.</p>
<pre class="bash"><code>#If you want to delete a user and its home directory and mail spool
sudo deluser --remove-home username</code></pre>
</div>
<div id="create-groups" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Create groups</h2>
<p>To create a group, here <code>genomics</code> type the following command:</p>
<pre class="bash"><code>#Create a group entitled `genomics`
sudo groupadd genomics</code></pre>
</div>
<div id="add-users-to-groups" class="section level2" number="3.4">
<h2><span class="header-section-number">3.4</span> Add users to groups</h2>
<p>To add users to the groups (here <code>genomics</code>) type the following command:</p>
<pre class="bash"><code>#Add users to the group, here `genomics`
sudo usermod -a -G genomics userID

# Examples
sudo usermod -a -G genomics svenbuerki
sudo usermod -a -G genomics bio_11
sudo usermod -a -G genomics anthonymelton</code></pre>
<p>To know which users are already members of a group type the following command:</p>
<pre class="bash"><code>grep genomics /etc/group</code></pre>
</div>
<div id="create-shared-folders" class="section level2" number="3.5">
<h2><span class="header-section-number">3.5</span> Create shared folders</h2>
<p>When multiple users need accessing to the same data, it is best practice to create shared folders and assign them to a group (made of multiple users, here the <code>genomics</code> group).</p>
<p>To create the folder to be shared (here <code>Genomics_shared</code>) type the following command:</p>
<pre class="bash"><code>#General syntax
sudo mkdir -p /path/Shared_Folder_ID

#Example (saved in home/)
sudo mkdir -p /home/Genomics_shared</code></pre>
</div>
<div id="assign-shared-folders-to-a-group" class="section level2" number="3.6">
<h2><span class="header-section-number">3.6</span> Assign shared folders to a group</h2>
<p>Here, we assign a group (here <code>genomics</code>) to become owner of the shared folder (here <code>/home/Genomics_shared</code>) and give them appropriate permissions, which are also inherited by the subdirectories. This is done as follows:</p>
<pre class="bash"><code>sudo chgrp -R genomics /home/Genomics_shared
sudo chmod -R 2775 /home/Genomics_shared</code></pre>
</div>
<div id="switch-user-to-check-changes" class="section level2" number="3.7">
<h2><span class="header-section-number">3.7</span> Switch user to check changes</h2>
<p>To switch user on Ubuntu you need to type the following command:</p>
<pre class="bash"><code>#General syntax
su USER_name

#Example (you will have to provide password for bio_11)
su bio_11</code></pre>
</div>
</div>
<div id="estimate-file-space-usage" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Estimate file space usage</h1>
<p>Since NGS data are taking a lot of space on the hard drive, users might would like to quickly estimate the amount of space used by their files in a folder. To do that, please use the following command relying on <code>du</code> protocol:</p>
<pre class="bash"><code>du -h -s *</code></pre>
</div>
<div id="add-symbolic-links-to-programs" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Add symbolic links to programs</h1>
<p>Here we use the <code>ln</code> command to make a symbolic link (or a kind of alias) to a program in the <code>bin/</code>. This means that the program will be accessible to anybody by typing a shortcut (or alias).</p>
<p>Since the symbolic link will reside in the <code>bin/</code> folder, first you will be navigating to this location as follows:</p>
<pre class="bash"><code>#Navigate where the symbolic link should reside
cd /usr/local/bin/</code></pre>
<p>Then we will establish the symbolic link as follows:</p>
<pre class="bash"><code>#Use ln protocol to establish symbolic link
# pathtoprogram = path to executable program 
# alias = the name you give to the symbolic link or alias
sudo ln -s pathtoprogram alias

#Example with R
sudo ln -s /home/bioinformatics/R-4.0.4/bin/R R-4</code></pre>
<p>In this example, everybody will be able to access <code>R</code> version 4.0.4 by typing <code>R-4</code> in the Terminal.</p>
</div>
<div id="mount-a-smb-server-on-ubuntu" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Mount a smb server on Ubuntu</h1>
<ol style="list-style-type: decimal">
<li>Create (<code>mkdir</code>) or identify a folder on your Ubuntu computer where the smb server will be connected to. In this example, we use our shared folder in <code>/home/Genomics_shared</code> and created a folder for smb protocols (<code>smbshare</code>).</li>
</ol>
<pre class="bash"><code>#Location where smb server will be mounted
/home/Genomics_shared/sambashare</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Mount the smb server using this command:</li>
</ol>
<pre class="bash"><code>#1. Mount smb server
# adjust: 
# - username
# - smb IP address
# - Location for mounting smb server on your computer
sudo mount -t cifs -o username=svenbuerki //cifs-prd-01/research/SvenBuerki /home/Genomics_shared/sambashare</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>You can copy or move files and folders freely from your local computer to smb server by using <code>sudo</code> protocol. Here is an example where we copy a folder from computer to mounted smb server:</li>
</ol>
<pre class="bash"><code>#2. Copy folder from Linux box to smb server
sudo cp -r IDT2-2_NovaSeq_Illumina_April2019/ /home/Genomics_shared/sambashare/</code></pre>
<p>If you are interested in setting-up a samba (smb) served from your Linux computer to be shared with Windows users follow this approach:</p>
<p><a href="https://ubuntu.com/tutorials/install-and-configure-samba#1-overview" class="uri">https://ubuntu.com/tutorials/install-and-configure-samba#1-overview</a></p>
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Loureiro2007" class="csl-entry">
Loureiro, João, Eleazar Rodriguez, Jaroslav Doležel, and Conceição Santos. 2007. <span>“<span class="nocase">Two New Nuclear Isolation Buffers for Plant DNA Flow Cytometry: A Test with 37 Species</span>.”</span> <em>Annals of Botany</em> 100 (4): 875–88. <a href="https://doi.org/10.1093/aob/mcm152">https://doi.org/10.1093/aob/mcm152</a>.
</div>
<div id="ref-Pellicer2014" class="csl-entry">
Pellicer, Jaume, and Ilia J. Leitch. 2014. <span>“The Application of Flow Cytometry for Estimating Genome Size and Ploidy Level in Plants.”</span> In <em>Molecular Plant Taxonomy: Methods and Protocols</em>, edited by Pascale Besse, 279–307. Totowa, NJ: Humana Press. <a href="https://doi.org/10.1007/978-1-62703-767-9_14">https://doi.org/10.1007/978-1-62703-767-9_14</a>.
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
