--- 
title: 'Protocols'
output:
  bookdown::html_document2: 
    toc: TRUE
    toc_depth: 4
    toc_float: TRUE
link-citations: yes
fontsize: 12pt
bibliography: References.bib
---

```{r packages, echo=FALSE, warning=FALSE, include=FALSE}
library(knitr)
library(rmarkdown)
library(bookdown)
#library(distill)
library(knitcitations)
library(formatR)
library(devtools)
library(usethis)
library(lubridate)
library(scales)

#Generate BibTex citation file for all R packages used to produce report
knitr::write_bib(.packages(), file = 'packages.bib')
```


```{r setup, include=FALSE, cache=FALSE, message = FALSE}
#Load packages
library(knitr)
library(rmarkdown)
# Chunk options: see http://yihui.name/knitr/options/ ###

## Text results
opts_chunk$set(echo = TRUE, warning = TRUE, message = TRUE, include = TRUE)

## Code decoration
#opts_chunk$set(tidy = TRUE, R.options = list(width = 60), highlight = TRUE)

## Caching code
opts_chunk$set(cache = 2, cache.path = "cache/")

## Plots
#opts_chunk$set(fig.path = "Figures_MS/", dev=c('pdf', 'png'), dpi = 300)

## Locate figures as close as possible to requested position (=code)
#opts_chunk$set(fig.pos = "H")
```


# Chromosome counting

## Objective

Counting chromosomes of *Artemisia tridentata* using root meristems of individual lines maintained *in vitro* at BSU. This approach will help ascertaining the ploidy level (2n) of our individual lines.

## Material & reagents

**Day 1**

- Little cooler for getting ice (room XXX).
- Labeled glass large Petri dishes.
- Tweezers (to pull roots out of Magenta vessels).
- Distilled water.
- Space in the fridge to store at 4<span>&deg;</span>C for at least 10 hours (can do 24 hours).


## Plant material

Young whitish roots are sampled from individuals cultured *in vitro* either at <span style="color: black;font-weight:bold">Growth</span> or <span style="color: blue;font-weight:bold">Rooting</span> phases. 

## Sampling and fixing tissue

1. Sample young whitish roots and place them in a Petri dish.
2. Use achromatic spindle inhibitors to stop mitotic divisions (at metaphase) and contract chromosomes. This can be done following two approaches: 
    - Treat roots with saturated solution of water and <span>&#x3B1;</span>-monobromonaphtalene for 3 hours at room temperature. Start by homogenizing the stock solution and adding supernatant to cover roots in the Petri dish. 
    - Treat roots in a mix of water and crushed ice (4<span>&deg;</span>C) for 10 hours.
3. Plant tissue fixation and conserving chromosomes. 
    - Prepare a falcon tube with the solution of ethanol:acetic acid (3:1). Add few drops of the acetocarmin solution and one drop of iron acetate. Homogenize the solution by gently shaking the falcon tube.
    - Transfer roots into falcon tube and store at room temperature for 1-2 days. 

## Staining tissue

4. Pour acetocarmin solution in a porcelain crucible and add roots. Cover crucible with a lid.
5. Prepare a Petri dish filled with 45% acetic acid. This will be used to cool down stained roots and preserve them until chromosome numbers is established.
6. Gently heat solution in porcelain crucible with a Bunsen burner for ca. 2 minutes. Avoid boiling! This will stain chromosomes in red.
7. Pour content of porcelain crucible into Petri dish.    

## Squashing tissue

8. Put a drop of the solution in the Petri dish on a microscope slide and add a root tip in it. 
9. Remove the root cap under the scope using tweezers.This will facilitate the squash and observation of chromosomes.
10. Cover with a plastic coverslip and use handle of tweezers to squash the tissue. Go gently! 
11. Observe under microscope and take picture to validate chromosome count.

## Preparing stock solutions

Recipes to prepare solutions presented here were adapted from this [document](Data/Caryologie_methodes_INRA_091118.pdf) (in french).

### Ethanol:acetic acid solution

- Mix one volume of acetic acid for three volumes of ethanol. Store at 4<span>&deg;</span>C.

### Acetocarmin solution

For 1000 mL for acetocarmin solution: 

1. Measure 450 mL of 45% glacial acetic acid and pour in **XXX** (a vessel that we can heat). See Figure in this [document](Data/Carmin_acétique_acétate_de_fer.pdf).
2. Measure 550 mL of distilled water and add to vessel.
3. Weigh 5 gr. of carmin (Fisher C579-25). Carmine is a basic dye that is prepared from the insect *Coccus cacti*.
4. Add boileezers, and reflux for 2-3 hours.
5. Filter into dark bottles and store at 4<span>&deg;</span>C.

This solution can be stored for a long time. Staining can be intensified by adding ferric chloride (FeCl~2~·6H~2~O); add 5 mL of a 10 % ferric chloride solution per 100 mL of % acetocarmine.

Kansas State University as also put together a slightly different protocol for the acetocarmin solution: https://www.k-state.edu/wgrc/electronic_lab/aceto_stain.html

# Flow cytometry

## Objective

Estimate 2C genome size of *Artemisia tridentata* based on leaf material using the one-step protocol with general purpose buffer (GPB) by @Loureiro2007 and the protocol described in @Pellicer2014. The publications are accessible [here](Data/). 

The flow cytometry will be conducted at BSU using a [BD Accuri™ C6 Flow Cytometer](Data/bd_accuri_optical_filter_guide.pdf).

## Challenge

One thing that we need to bear in mind is that the coefficient of variation of the fluorescence peaks should be below 5%, and that the measurements should be taken using a linear scale rather that a logarithmic scale.

## Plant material

Genome size is estimated using **young, fresh leaves for both our target (sagebrush) and internal standards**. 

Concerning reference internal standards, we recommend that several species, covering a broad range of genome sizes, are kept growing in the laboratory to enable the most appropriate standard to be selected for each particular analysis.

### Internal standards

Check what seeds we have in the lab.:

- *Petroselinum crispum* ‘Curled moss’ (2C=4.5 pg).
- *Pisum sativum* ‘Ctirad’ (2C=9.09 pg).
- *Solanum lycopersicum* L. ‘Stupiké polní rane’ (2C=2 pg).


## Protocol

The recipe of the buffer and the steps are described in the paper, but briefly:

- Work under cold conditions (keep samples and buffers on ice).
- Cut a small portion of leaf tissue (1cm^2^, or less) from both the target and the internal standard.
- Place samples in a small petri dish with 1ml of ice-cold buffer.
- Chop with a sharp new razor blade till obtaining a fine lysate.
- Add another 1ml of buffer.
- Filter through a 30 um nylon mesh.
- Stain with 100 ul of 1mg/ml Propidium iodide. Nuclei were stained with 50 µg mL−1 propidium iodide (PI; Fluka, Buchs, Switzerland), and 50 µg mL−1 RNase (Sigma, St Louis, MO, USA) was added to nuclear suspension to prevent staining of double-stranded RNA.
- Incubate 5-10 min on ice.
- Measure on the flow cytometer (here the [BD Accuri™ C6 Flow Cytometer](Data/bd_accuri_optical_filter_guide.pdf)).

## Preparing buffer

### General composition of the lysis buffer

**GBP:** 0.5 mM spermine.4HCl, 30 mM sodium citrate, 20 mM MOPS (MOPS, 4-Morpholinepropane sulfonate), 80 mM KCl, 20 mM NaCl, 0.5 % (v/v) Triton X-100, pH 7.0.

### Procedure to prepare the lysis buffer (GBP)

1. Weigh the following chemicals into a glass (final volume of 200 mL):
    - Spermine 0.5 mM (0.0348 g)
    - Sodium citrate 3mM (1.764 g)
    - MOPS 20 mM (0.836 g)
    - KCl 80 mM (1.193 g)
    - NaCl 20 mM (0.233 g)
2. Add double distilled water to 200 mL and mix with a rotating magnet and dissolve the chemicals
3. Add 1 mL of Triton X-100 and mix until diluted
4. Adjust to pH 7
5. Add 6 g of PVP-40 and mix until dissolved (Add PVP while the buffer is being mixed as it tends to clog. Be patient and leave it with a rotating magnet until completely dissolved)
6. Store in the fridge or prepare aliquots of 50 mL which can be stored on the freezer. Use one each time and once thawed do not freeze again. It can be stored in the fridge for weeks.

The buffer is now ready to use. If you notice that the sample becomes dark when prepared, add 200 ul of B-mercaptoethanol to the buffer and mix (or equivalent if aliquots have been prepared).

# Managing Linux accounts and sharing folders

To more efficiently benefit from our Linux computers, we will provide overview of protocols to:

- Create users.
- Delete users.
- Create groups.
- Add users to groups.
- Create shared folders.
- Assign shared folders to a group.
- Switch user to check changes.

In RStudio, you can send commands directly into the Terminal (instead of the Console) by using this combination of keys: `Ctrl + Alt + Enter`. The Terminal has to be selected before using this approach.

## Create users

For this purpose, we will be using the `adduser` command:

```{bash, eval=F}
#Create user, add password and create home directory (without admin rights)
sudo adduser username

#For example
sudo adduser anthonymelton
```

You will be asked to answer multiple questions, only the password is mandatory.

Complete creation of sub-directories for new user by typing these commands:

```{bash, eval=F}
#Switch to the new user account (see below for more details)
su username

#Create sub-directories (e.g. Documents/, Downloads/, etc.)
xdg-user-dirs-update --force
```

To view the list of users on the Linux computer type this command:

```{bash, eval=F}
cut -d : -f 1 /etc/passwd
```

To view the list of logged in users do:

```{bash, eval=F}
w
```

To **provide admin (`sudo`) rights to a user** that you created do the following:

```{bash, eval=F}
#Create a new used with admin/sudo rights
sudo usermod -aG sudo username

#For example
sudo usermod -aG sudo anthonymelton
```

On the other hand, **if you want to reverse a user from `sudo` to `standard` rights**, do the following:

```{bash, eval=F}
#General syntax
sudo deluser username sudo
```

This command will only remove user 'username' from the sudo group, but it will not delete the user permanently from the system.  Now, they become regular user and cannot do any administrative tasks as sudo user.

## Delete users

After someone leaves the lab, you might have to delete their user account. You can do that by using the following command, but please make sure that you have archived all their data and code prior to proceeding.

```{bash, eval=F}
#If you want to delete a user and its home directory and mail spool
sudo deluser --remove-home username
```

## Create groups

To create a group, here `genomics` type the following command:

```{bash, eval=F}
#Create a group entitled `genomics`
sudo groupadd genomics
```

## Add users to groups

To add users to the groups (here `genomics`) type the following command:

```{bash, eval=F}
#Add users to the group, here `genomics`
sudo usermod -a -G genomics userID

# Examples
sudo usermod -a -G genomics svenbuerki
sudo usermod -a -G genomics bio_11
sudo usermod -a -G genomics anthonymelton
```

To know which users are already members of a group type the following command:

```{bash, eval=F}
grep genomics /etc/group
```

## Create shared folders

When multiple users need accessing to the same data, it is best practice to create shared folders and assign them to a group (made of multiple users, here the `genomics` group). 

To create the folder to be shared (here `Genomics_shared`) type the following command:

```{bash, eval=F}
#General syntax
sudo mkdir -p /path/Shared_Folder_ID

#Example (saved in home/)
sudo mkdir -p /home/Genomics_shared
```

## Assign shared folders to a group

Here, we assign a group (here `genomics`) to become owner of the shared folder (here `/home/Genomics_shared`) and give them appropriate permissions, which are also inherited by the subdirectories. This is done as follows:

```{bash, eval=F}
sudo chgrp -R genomics /home/Genomics_shared
sudo chmod -R 2775 /home/Genomics_shared
```

## Switch user to check changes

To switch user on Ubuntu you need to type the following command:

```{bash, eval=F}
#General syntax
su USER_name

#Example (you will have to provide password for bio_11)
su bio_11
```

# Add symbolic links to programs

Here we use the `ln` command to make a symbolic link (or a kind of alias) to a program in the `bin/`. This means that the program will be accessible to anybody by typing a shortcut (or alias). 

Since the symbolic link will reside in the `bin/` folder, first you will be navigating to this location as follows:

```{bash, eval=F}
#Navigate where the symbolic link should reside
cd /usr/local/bin/
```

Then we will establish the symbolic link as follows:

```{bash, eval=F}
#Use ln protocol to establish symbolic link
# pathtoprogram = path to executable program 
# alias = the name you give to the symbolic link or alias
sudo ln -s pathtoprogram alias

#Example with R
sudo ln -s /home/bioinformatics/R-4.0.4/bin/R R-4
```

In this example, everybody will be able to access `R` version 4.0.4 by typing `R-4` in the Terminal.

# Disk status

Since NGS data are taking a lot of space on the hard drive, users might would like to quickly estimate the amount of space left on their hard drive (using `df`) or space taken by files (using `du`). To do that, please use the following commands: 

## Disk space

The `df` protocol gives you insights into disk free space. To know about the free space on all disks do:

```{bash, eval=F}
df -H
```

To get the same information, but for the disk specific to a folder (here `/home`), do:

```{bash, eval=F}
df -H /home
```

## Disk usage

The `du` protocol gives you insights into disk usage. here we propose a command that gives you specifically the disk usage taken by files:

```{bash, eval=F}
#Disk usage taken by files in a folder
du -h -s *
```

# Share external hard drive with all users

If you want to share an external hard drive across all users of a Ubuntu computer you will need to add an entry in the `fstab` file. However, before adding this entry, you will need to:

- Identify the `UUID` of the external hard drive (using `sudo blkid`). If your hard drive uses a exFAT file system you will have to install additional packages to mount it on Linux (see below).
- Create a folder owned by root (and give writing permissions  to all users) to mount the external hard drive.

To share external hard drive with all users follow the following procedure:

1. Identify the `UUID` of the external hard drive:
```{bash, eval=F}
#Find UUID of your device (make sure that it is plugged in before running this command)
sudo blkid
```

**WARNING:** Certain hard drives (e.g. WD) have a different data encryption mode (or file system), which are not directly compatible with Linux and you won't be able to mount them (even if you see their `UUID`. For instance, the exFAT file system is ideal for flash drives and SD cards, but you will need to install a few packages to be able to mount it. Open the Terminal and run this command:
```{bash, eval=F}
sudo apt-get install exfat-fuse exfat-utils
```

You should now be able to see the hard drive and mount it. To share the hard drive with all users you will have to edit the `fstab` file in a different manner (see below).

2. Create a folder owned by root in `/media`:
```{bash, eval=F}
#Create a folder in media (adjust name, here SeaGate)
sudo mkdir /media/SeaGate
```

3. Give all users permissions to read and write (in `/media/SeaGate`):
```{bash, eval=F}
sudo chmod -R 777 /media/SeaGate
```

4. Edit and save the `fstab` file:
```{bash, eval=F}
#Open file
sudo vim /etc/fstab

#Add entry (adjust UUID to your UUID) at bottom of file
# Separate blocks with tab
UUID=00148BDE148BD4D6 /media/SeaGate/    ntfs-3g        auto,user,rw 0 0

#For exFAT file system do
UUID=00148BDE148BD4D6 /media/WD/    exfat-fuse        auto,user,rw 0 0
```

5. Reboot system and you should be good to go.

# Install and configure samba

A Samba file server enables file sharing across different operating systems over a network. It lets you access your desktop files from a laptop and share files with Windows and macOS users. Here, we will be using this protocol to share our external hard drive remotely via  smb protocol.

## Installing Samba

If it is not done yet, please install Samba on Ubuntu computer as follows:

```{bash, eval=F}
sudo apt update
sudo apt install samba
```

We can check if the installation was successful by running:
```{bash, eval=F}
whereis samba
```

## Sharing folder with Samba protocol

Now that Samba is installed, we can share a folder, here `/media/SeaGate`.

To share the folder, you have to edit the configuration file for Samba located at `/etc/samba/smb.conf`. To add the new directory as a share, we edit the file by running:
```{bash, eval=F}
sudo vim /etc/samba/smb.conf
```

Then add the following text at the bottom of the file:
```{bash, eval=F}
[SeaGate]
    comment = Samba on Ubuntu
    path = /media/SeaGate
    read only = no
    browsable = yes
```

Now that we have our new share configured, save it and restart Samba for it to take effect:
```{bash, eval=F}
sudo service smbd restart
```

Update the firewall rules to allow Samba traffic:
```{bash, eval=F}
sudo ufw allow samba
```

## Setting up User Accounts and Connecting to Share

Since Samba doesn’t use the system account password, we need to set up a Samba password for our user account:
```{bash, eval=F}
#General syntax
sudo smbpasswd -a username
#Example  for us
sudo smbpasswd -a anthonymelton
```

## Connecting to Share

You can now access this folder remotely by using `smb` protocol. 

On macOS: In the Finder menu, click Go > Connect to Server then enter:
```{bash, eval=F}
#General syntax (where share =  the name of the share)
smb://ip-address/share

#Example
smb://132.178.142.223/SeaGate
```

# Mount a smb server on Ubuntu

1. Create (`mkdir`) or identify a folder on your Ubuntu computer where the smb server will be connected to. In this example, we use our shared folder in `/home/Genomics_shared` and created a folder for smb protocols (`smbshare`).
```{bash, eval=F}
#Location where smb server will be mounted
/home/Genomics_shared/sambashare
```

2. Mount the smb server using this command:
```{bash, eval=F}
#1. Mount smb server
# adjust: 
# - username
# - smb IP address
# - Location for mounting smb server on your computer
sudo mount -t cifs -o username=svenbuerki //cifs-prd-01/research/SvenBuerki /home/Genomics_shared/sambashare
```
3. You can copy or move files and folders freely from your local computer to smb server by using `sudo` protocol. Here is an example where we copy a folder from computer to mounted smb server:
```{bash, eval=F}
#2. Copy folder from Linux box to smb server
sudo cp -r IDT2-2_NovaSeq_Illumina_April2019/ /home/Genomics_shared/sambashare/
```

If you are interested in setting-up a samba (smb) served from your Linux computer to be shared with Windows users follow this approach:

https://ubuntu.com/tutorials/install-and-configure-samba#1-overview

# SFTP protocol

You might have to download NGS data from different providers (e.g. GeneWIZ) and they will set-up an `sftp` server for you. Here is a short tutorial on the procedure used to download data:

## Start a tmux session

Open a Terminal and start a tmux session as follows:

```{bash, eval=F}
tmux
```

This will open a safe working environment that can be accessed very easily. Please read protocol on `tmux` sessions [here](https://svenbuerki.github.io/Genomics-Bioinformatics/Tutorials.html#3_Running_bioinformatic_analyses_-_tmux).

## Connect to sftp server

To connect to an sftp server you need:

- username (here anthonymelton_boisestate).
- sftp IP server address (here sftp.genewiz.com)

```{r, eval=F}
#General syntax
sftp username@sftp.IP

#Example with GeneWIZ
sftp anthonymelton_boisestate@sftp.genewiz.com
```

## Set destination path

The `get` or `mget` protocols used to copy files over don't allow to set the destinations path (where the files will be saved on your local computer), but you can easily set this by using the `lcd` command. Here is an example based on our project structure.

```{bash, eval=F}
#Copy files from sftp server to /home/Genomics_shared/G1_b2_1/
lcd /home/Genomics_shared/G1_b2_1/
```

## Copy whole project folder

The best practice is to copy the whole project folder over onto your local computer. This can be done by using `get -r` command. Here is an example with some GeneWIZ data.

```{bash, eval=F}
#Copy whole project over
get -r 30-470086959
```

Depending on the sive of the files (usually several Gb)
it might take hours for the transfer to be completed. be patient!

# References

