<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Genome sequencing, phasing and annotation</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Sagebrush Genome Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Individual_lines.html">Propagation</a>
</li>
<li>
  <a href="Biomass_production.html">Biomass</a>
</li>
<li>
  <a href="Sequencing.html">Genome</a>
</li>
<li>
  <a href="Protocols.html">Protocols</a>
</li>
<li>
  <a href="Team.html">Team</a>
</li>
<li>
  <a href="funding.html">Acknowledgements</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Genome sequencing, phasing and annotation</h1>

</div>


<div id="overview" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Overview</h1>
<p>Summaries of procedures applied in this project to sequence, phase and annotate the sagebrush genome based on the <code>G2_b24_1</code> individual line is presented here. Click <a href="Individual_lines.html">here</a> to learn more about <code>G2_b24_1</code>. We are briefly presenting below our predicted sequence data and biomass requirements for each sequencing technology.</p>
<div id="sequence-data" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Sequence data</h2>
<p>Our sequencing and assembly strategy is described in section <a href="#wetlab">2</a>, but a summary of sequence data and their associated NGS platforms is provided below (see also Table <a href="#tab:tabproject">1.1</a>):</p>
<ul>
<li><a href="https://www.ncbi.nlm.nih.gov/genbank/wgs/"><strong>Whole Genome Shotgun</strong></a>:
<ul>
<li>Illumina HiSeq platform: To infer genome size and complexity (and produce a haploid draft genome).</li>
<li>PacBio Sequel II (using 8M SMRT cells): To conduct <em>de novo</em> assembly.</li>
</ul></li>
<li><strong>Proximity ligation</strong>:
<ul>
<li><a href="https://dovetailgenomics.com/intro-omni-c-tech/">Omni-C libraries</a> (sequenced on and Illumina platform): To conduct phasing.</li>
</ul></li>
<li><strong>RNASeq</strong>:
<ul>
<li>Illumina NovaSeq platform: To perform genome annotation.</li>
</ul></li>
<li><a href="https://www.ncbi.nlm.nih.gov/sra/"><strong>SRA Experiments</strong></a>: Sequence Read Archive (SRA) data will be made available on NCBI upon submission of the genome.</li>
<li><strong>Protein Sequences</strong>: Protein sequences will be made available.</li>
<li><strong>Assembly/Genome</strong>: An assembly at pseudo-chromosome level will be made available on NCBI upon submission of the genome.</li>
</ul>
</div>
<div id="biomass-requirements-per-sequencing-technology" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Biomass requirements per sequencing technology</h2>
<p>We are estimating that ca. 120 gr of leaf biomass are necessary for genome sequencing, phasing and annotation. Table <a href="#tab:tabproject">1.1</a> provides a summary of sequencing technologies applied in this project, their purpose and biomass requirements. These data do not account for a preliminary DNA extraction trial (to conduct PacBio sequencing).</p>
<table class="table table-striped table-hover table-condensed" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:tabproject">Table 1.1: </span>Summary of sequencing technologies, their purpose and requirements to complete the sagebrush genome project.
</caption>
<thead>
<tr>
<th style="text-align:left;">
Type
</th>
<th style="text-align:left;">
Purpose
</th>
<th style="text-align:left;">
Unit/Biomass
</th>
<th style="text-align:left;">
Total units / biomass
</th>
<th style="text-align:right;">
Number of plantlets
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Illumina HiSeq
</td>
<td style="text-align:left;">
Genome size and complexity (incl. haploid draft genome)
</td>
<td style="text-align:left;">
1 Illumina library = 20 mg
</td>
<td style="text-align:left;">
1 library = 20 mg
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
PacBio sequencing
</td>
<td style="text-align:left;">
<em>De novo</em> genome assembly
</td>
<td style="text-align:left;">
1 cell = 20 gr
</td>
<td style="text-align:left;">
5 cells = 100 gr
</td>
<td style="text-align:right;">
125
</td>
</tr>
<tr>
<td style="text-align:left;">
Proximity ligation (Hi-C for phasing genome)
</td>
<td style="text-align:left;">
Phasing
</td>
<td style="text-align:left;">
1 Illumina library = 6 gr
</td>
<td style="text-align:left;">
3 libraries = 18 gr
</td>
<td style="text-align:right;">
23
</td>
</tr>
<tr>
<td style="text-align:left;">
RNA-seq
</td>
<td style="text-align:left;">
Annotation
</td>
<td style="text-align:left;">
1 library = 20 mg
</td>
<td style="text-align:left;">
1 library = 20 mg
</td>
<td style="text-align:right;">
1
</td>
</tr>
</tbody>
</table>
<p>In the case of the Illumina HiSeq, although we will only produce 1 library, it will be dispatched on five HiSeq runs.</p>
</div>
<div id="sequencing-data-by-technology" class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> Sequencing data by technology</h2>
<p>The amount of data (bp) produced per sequencing technology is provided in Table <a href="#tab:dataseq">1.2</a> together with estimates of haploid genome coverage (x).</p>
<table class="table table-striped table-hover table-condensed" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:dataseq">Table 1.2: </span>Summary of sequencing data produced by technology for the sagebrush genome project. Haploid genome size = 4.5Gbp
</caption>
<thead>
<tr>
<th style="text-align:left;">
Type
</th>
<th style="text-align:left;">
Purpose
</th>
<th style="text-align:right;">
Data (bp)/Run
</th>
<th style="text-align:right;">
N. runs
</th>
<th style="text-align:right;">
Total data (bp)
</th>
<th style="text-align:right;">
Haploid genome coverage (x)
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Illumina HiSeq
</td>
<td style="text-align:left;">
Genome size and complexity (incl. haploid draft genome)
</td>
<td style="text-align:right;">
1.05e+11
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
5.25e+11
</td>
<td style="text-align:right;">
116.67
</td>
</tr>
<tr>
<td style="text-align:left;">
PacBio sequencing
</td>
<td style="text-align:left;">
<em>De novo</em> genome assembly
</td>
<td style="text-align:right;">
5.00e+10
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
2.50e+11
</td>
<td style="text-align:right;">
55.56
</td>
</tr>
<tr>
<td style="text-align:left;">
Proximity ligation (Hi-C for phasing genome)
</td>
<td style="text-align:left;">
Phasing
</td>
<td style="text-align:right;">
3.00e+10
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
9.00e+10
</td>
<td style="text-align:right;">
20.00
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="wetlab" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Wet-lab procedures</h1>
<p>Completion of the wet-lab work by Dovetail Genomics detailed here is predicted to take 28 weeks (7 months) upon receipt of the biomass. To gain insights into the timetable for biomass production, please click <a href="Biomass_production.html">here</a>.</p>
<ol style="list-style-type: decimal">
<li>Ascertain the ploidy level, genome size and genome complexity of <code>G2_b24_1</code>. This will be done by counting chromosomes (based on root squashes), inferring 2C genome size using flow cytometry (based on root and leaf tissues) and estimating genome size and complexity by applying a k-mer approach on Illumina HiSeq data.</li>
<li>Dovetail Genomics will extract HMW DNA with an average fragment size ca. 50-100kb based on biomass supplied by the team (see <a href="Biomass_production.html">Biomass production</a>).</li>
<li>Whole genome sequencing (WGS) will be conducted using PacBio technology. We aim at generating 50-100X raw data coverage (haploid sagebrush genome size: 4.5 Gbp). To achieve this objective, 5 PacBio CLR (continuous long-read) libraries will be built and sequenced on the PacBio Sequel II using 8M SMRT cells. A SMRT cell should generate &gt;50Gb of raw data. In our case, 250Gb (5x 50Gb) of raw data will be generated, which corresponds to 55X haploid genome coverage (we are on the low side, but it should be fine). These data will be used to assemble the haploid draft genome (used as input for phasing based on Omni-C libraries).</li>
<li>Three <a href="https://dovetailgenomics.com/intro-omni-c-tech/">Omni-C libraries</a> (= proximity ligation libraries) will be constructed (one library per 3 Gbp of the organism’s genome) based on the HMW DNA extractions. These libraries will be sequenced on an Illumina HiSeq X instrument (ca. 100M PE150bp read pairs per Gbp of genome size). Before final sequencing, Dovetail will conduct some shallow sequencing (ca. 2M PE75 bp) to assess library quality.</li>
<li>RNA will be extracted from leaf tissue (20mg) by Dovetail and a standard RNA library will be prepared with rRNA-depletion. 2x150bp read will be sequenced on an Illumina platform and raw data from ca. 20M read pairs per sample (1 in this project) will be used for genome annotation.</li>
</ol>
</div>
<div id="bioinformatic-procedures" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Bioinformatic procedures</h1>
<ol style="list-style-type: decimal">
<li>PacBio <em>de novo</em> assembly (= produce haploid draft assembly). We aim at a minimum N50 of 100kb (needed for HiRise pipeline).</li>
<li>The haploid daft assembly will be scaffolded (phased) through the HiRise software pipeline using the proximity ligation data (this software is owned by Dovetail). The Omni-C libraries can be used for genome assembly and haplotype phasing. We will also be able to use this data to call SNPs and look at structural variations.</li>
<li>Whole genome annotation will be conducted using RNA-seq data. The basic services will be included:
<ol style="list-style-type: lower-alpha">
<li>Repeat masking.</li>
<li><em>Ab initio</em> gene prediction using related species (e.g. <em>Artemisia annua</em>).</li>
<li>RNAseq mapping to enhance annotation.</li>
<li>Manual curation of 5 genes (not very interesting for us). The work done by Anthony Melton based on seedlings from GxE experiment will be key here.</li>
<li>Assignment of functional tags to genes.</li>
</ol></li>
<li>Results files will be delivered electronically via secure FTP.</li>
</ol>
<div id="refs">

</div>
</div>
<div id="appendix-1" class="section level1 appendix" number="4">
<h1><span class="header-section-number">4</span> Appendix 1</h1>
<p>Citations of all R packages used to generate this report.</p>
<p>[1] J. Allaire, Y. Xie, J. McPherson, et al. <em>rmarkdown: Dynamic
Documents for R</em>. R package version 2.6. 2020. &lt;URL:
<a href="https://github.com/rstudio/rmarkdown" class="uri">https://github.com/rstudio/rmarkdown</a>&gt;.</p>
<p>[2] C. Boettiger. <em>knitcitations: Citations for Knitr Markdown Files</em>.
R package version 1.0.10. 2019. &lt;URL:
<a href="https://github.com/cboettig/knitcitations" class="uri">https://github.com/cboettig/knitcitations</a>&gt;.</p>
<p>[3] R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2019. &lt;URL: <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>&gt;.</p>
<p>[4] H. Wickham and J. Bryan. <em>usethis: Automate Package and Project
Setup</em>. R package version 2.0.0. 2020. &lt;URL:
<a href="https://CRAN.R-project.org/package=usethis" class="uri">https://CRAN.R-project.org/package=usethis</a>&gt;.</p>
<p>[5] H. Wickham, J. Hester, and W. Chang. <em>devtools: Tools to Make
Developing R Packages Easier</em>. R package version 2.3.2. 2020. &lt;URL:
<a href="https://CRAN.R-project.org/package=devtools" class="uri">https://CRAN.R-project.org/package=devtools</a>&gt;.</p>
<p>[6] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. ISBN 978-1138700109. Boca Raton, Florida: Chapman and
Hall/CRC, 2016. &lt;URL: <a href="https://github.com/rstudio/bookdown" class="uri">https://github.com/rstudio/bookdown</a>&gt;.</p>
<p>[7] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. R package version 0.21. 2020. &lt;URL:
<a href="https://github.com/rstudio/bookdown" class="uri">https://github.com/rstudio/bookdown</a>&gt;.</p>
<p>[8] Y. Xie. <em>Dynamic Documents with R and knitr</em>. 2nd. ISBN
978-1498716963. Boca Raton, Florida: Chapman and Hall/CRC, 2015. &lt;URL:
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>&gt;.</p>
<p>[9] Y. Xie. <em>formatR: Format R Code Automatically</em>. R package version
1.7. 2019. &lt;URL: <a href="https://github.com/yihui/formatR" class="uri">https://github.com/yihui/formatR</a>&gt;.</p>
<p>[10] Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in
R.” In: <em>Implementing Reproducible Computational Research</em>. Ed. by V.
Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and
Hall/CRC, 2014. &lt;URL:
<a href="http://www.crcpress.com/product/isbn/9781466561595" class="uri">http://www.crcpress.com/product/isbn/9781466561595</a>&gt;.</p>
<p>[11] Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.30. 2020. &lt;URL:
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>&gt;.</p>
<p>[12] Y. Xie, J. Allaire, and G. Grolemund. <em>R Markdown: The Definitive
Guide</em>. ISBN 9781138359338. Boca Raton, Florida: Chapman and Hall/CRC,
2018. &lt;URL: <a href="https://bookdown.org/yihui/rmarkdown" class="uri">https://bookdown.org/yihui/rmarkdown</a>&gt;.</p>
<p>[13] Y. Xie, C. Dervieux, and E. Riederer. <em>R Markdown Cookbook</em>. ISBN
9780367563837. Boca Raton, Florida: Chapman and Hall/CRC, 2020. &lt;URL:
<a href="https://bookdown.org/yihui/rmarkdown-cookbook" class="uri">https://bookdown.org/yihui/rmarkdown-cookbook</a>&gt;.</p>
</div>
<div id="appendix-2" class="section level1 appendix" number="5">
<h1><span class="header-section-number">5</span> Appendix 2</h1>
<p>Version information about R, the operating system (OS) and attached or R loaded packages. This appendix was generated using <code>sessionInfo()</code>.</p>
<pre><code>## R version 3.6.1 (2019-07-05)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS Mojave 10.14.6
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] formattable_0.2.0.1      leaflet_2.0.3            googlesheets4_0.2.0     
##  [4] kableExtra_1.2.1         dplyr_1.0.2              kfigr_1.2               
##  [7] MASS_7.3-53              forcats_0.5.0            TreeTools_1.4.0         
## [10] ggridges_0.5.2           stringr_1.4.0            ape_5.4-1               
## [13] ggtree_2.0.4             ggpubr_0.4.0             ggplot2_3.3.3           
## [16] chisq.posthoc.test_0.1.2 DT_0.16                  lsmeans_2.30-0          
## [19] emmeans_1.5.2-1          lmtest_0.9-38            zoo_1.8-8               
## [22] pscl_1.5.5               RColorBrewer_1.1-2       gplots_3.1.0            
## [25] scales_1.1.1             lubridate_1.7.9.2        devtools_2.3.2          
## [28] usethis_2.0.0            formatR_1.7              knitcitations_1.0.10    
## [31] bookdown_0.21            rmarkdown_2.6            knitr_1.30              
## 
## loaded via a namespace (and not attached):
##   [1] readxl_1.3.1        backports_1.2.1     fastmatch_1.1-0    
##   [4] plyr_1.8.6          igraph_1.2.6        lazyeval_0.2.2     
##   [7] crosstalk_1.1.0.1   digest_0.6.27       htmltools_0.5.0    
##  [10] fansi_0.4.1         magrittr_2.0.1      memoise_1.1.0      
##  [13] openxlsx_4.2.2      remotes_2.2.0       R.utils_2.10.1     
##  [16] askpass_1.1         prettyunits_1.1.1   colorspace_2.0-0   
##  [19] rvest_0.3.6         haven_2.3.1         rbibutils_1.4      
##  [22] xfun_0.20           callr_3.5.1         crayon_1.3.4       
##  [25] jsonlite_1.7.2      phangorn_2.5.5      glue_1.4.2         
##  [28] gargle_0.5.0        gtable_0.3.0        webshot_0.5.2      
##  [31] R.cache_0.14.0      car_3.0-10          pkgbuild_1.2.0     
##  [34] abind_1.4-5         mvtnorm_1.1-1       bibtex_0.4.2.3     
##  [37] rstatix_0.6.0       Rcpp_1.0.5          viridisLite_0.3.0  
##  [40] xtable_1.8-4        tidytree_0.3.3      foreign_0.8-75     
##  [43] bit_4.0.4           htmlwidgets_1.5.3   httr_1.4.2         
##  [46] ellipsis_0.3.1      pkgconfig_2.0.3     R.methodsS3_1.8.1  
##  [49] tidyselect_1.1.0    rlang_0.4.10        munsell_0.5.0      
##  [52] cellranger_1.1.0    tools_3.6.1         cli_2.2.0          
##  [55] generics_0.1.0      broom_0.7.1         evaluate_0.14      
##  [58] yaml_2.2.1          RefManageR_1.2.12   processx_3.4.5     
##  [61] bit64_4.0.5         fs_1.5.0            zip_2.1.1          
##  [64] caTools_1.18.0      purrr_0.3.4         nlme_3.1-149       
##  [67] R.oo_1.24.0         xml2_1.3.2          compiler_3.6.1     
##  [70] rstudioapi_0.13     curl_4.3            testthat_3.0.1     
##  [73] ggsignif_0.6.0      treeio_1.10.0       tibble_3.0.4       
##  [76] stringi_1.5.3       highr_0.8           ps_1.5.0           
##  [79] desc_1.2.0          lattice_0.20-41     Matrix_1.2-18      
##  [82] vctrs_0.3.6         pillar_1.4.7        lifecycle_0.2.0    
##  [85] BiocManager_1.30.10 Rdpack_2.1          estimability_1.3   
##  [88] data.table_1.13.6   bitops_1.0-6        gbRd_0.4-11        
##  [91] R6_2.5.0            KernSmooth_2.23-17  rio_0.5.16         
##  [94] codetools_0.2-16    sessioninfo_1.1.1   gtools_3.8.2       
##  [97] assertthat_0.2.1    pkgload_1.1.0       openssl_1.4.3      
## [100] rprojroot_2.0.2     withr_2.3.0         parallel_3.6.1     
## [103] hms_0.5.3           quadprog_1.5-8      grid_3.6.1         
## [106] tidyr_1.1.2         coda_0.19-4         rvcheck_0.1.8      
## [109] carData_3.0-4       googledrive_1.0.1</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
